---
title: "Progress report 3"
bibliography: ../../library/imap.bib
date: 'Updated: `r Sys.time()`'
output:
  html_document:
    css: ../../code/css/report.css
    # keep_md: yes
link-citations: yes
subtitle: 'Sequence Processing and classification'
biblio-style: apalike
---

<br>

```{r setup, include=FALSE}

load("./packages.RData")
load("./globalSetup.RData")
load("./metadatavars.RData")

knitr::opts_chunk$set(fig.path='./figures/', fig.show='asis',  dev = 'png', fig.align='center', out.width = "80%",  out.height = "80%", echo=TRUE, comment = NA, warning=FALSE, message=FALSE)
# knitr::opts_chunk$set(fig.path='./images/pdf/', fig.show='asis',  dev = 'pdf', fig.align='center', out.width = "80%",  out.height = "80%", echo=TRUE, comment = NA, warning=FALSE, message=FALSE)
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(comment = NA)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning = FALSE)

# Create required folders to be certain that they exist
dir.create("../../results/")
dir.create("../../results/tables/")
dir.create("../../results/tables/phylotype")

```

```{r message=FALSE, warning=FALSE}

# samplemetadata <- readr::read_table2(file="../../resources/metadata/samplemetadata.tsv", col_names = T)
# colnames(samplemetadata)[1] <- "SampleID"
# 
# # Subsetting metadata for mapping to output from dowmstream analysis
# library(dplyr)
# variables <- samplemetadata %>% dplyr::select("SampleID", "origID", "Var1", "Var2", "DayID", "DPW")
# names(variables)[2] <- "origID"
```

```{r mergereportdata, message=FALSE, warning=FALSE, include=FALSE}
library(readr)

# assembly report
contigrpt <- readr::read_table2(file = "../../data/mothur/qced.contigs.report")
contigrpt_2 <- contigrpt[c(2,3,6)]
names(contigrpt_2) <- c("SequenceLength","OverlapLength","MisMatches")

# alignment report
alignrpt <- read.table(file = "../../data/mothur/qced.trim.contigs.good.unique.align.report", header = TRUE, sep = "\t")
names(alignrpt) <- c("QueryName", "QueryLength", "TemplateName", "TemplateLength", "SearchMethod", "SearchScore", "AlignmentMethod", "QueryStart", "QueryEnd", "TemplateStart", "TemplateEnd", "PairwiseAlignmentLength", "GapsInQuery", "GapsInTemplate", "LongestInsert", "SimBtwnQueryTemplate")

alignrpt_2 <- alignrpt[c(2,12,16)]
names(alignrpt_2) <- c("QueryLength","AlignmentLength","PercentIdentity")

save(contigrpt, contigrpt_2, alignrpt, alignrpt_2, file = "../../results/mergealignreport.RData")

# names(alignrpt)
```

## Descriptive statistics of assembled sequence
```{r message=FALSE, warning=FALSE}
library(dplyr)
library(kableExtra)

load("../../results/mergealignreport.RData")

contigrpt_2_summary <- summary(contigrpt_2)
contigrpt_2_summary %>% knitr::kable() %>% kable_styling(bootstrap_options = "bordered", full_width = FALSE)
```

<br>
<br>

## Features of assembled sequences
```{r fig.height=4, fig.width=10}
library(ggplot2)
library(dplyr)

# Barplot
seqlen <- contigrpt_2 %>%
  ggplot( aes(x=SequenceLength)) +
    geom_bar(stat="count", fill="blue4", binwidth = 50) +
    theme_grey()  + 
    labs(x = "Sequence Length", y = "Sequences Count") +
  # xbreaks8 +
  ybreaks10 +
  formatPlot


overlaplen <- contigrpt_2 %>%
  ggplot( aes(x=OverlapLength)) +
    geom_bar(stat="count", fill="blue4", binwidth =25) +
    theme_grey()  + 
    labs(x = "Overlap Length", y = "Sequences Count") +
  # xbreaks8 +
  ybreaks10 +
  formatPlot

mismatch <- contigrpt_2 %>%
  ggplot( aes(x=MisMatches)) +
    geom_bar(stat="count", fill="blue4", binwidth = 10) +
    theme_grey()  +
    labs(x = "Number of Mismatch", y = "Sequences Count") +
  # xbreaks8 +
  ybreaks10 +
  formatPlot

ggpubr::ggarrange(seqlen, overlaplen, mismatch, ncol = 3, nrow = 1, align = "hv", labels = "AUTO", font.label = list(size = 12, color = "black", face = "bold", family = NULL))
```

<br>

## Descriptive statistics of aligned sequence

```{r message=FALSE, warning=FALSE}
alignrpt_2_summary <- summary(alignrpt_2)
alignrpt_2_summary %>% knitr::kable() %>% kable_styling(bootstrap_options = "bordered", full_width = FALSE)
```

<br>

## Features of aligned sequences

```{r align, fig.height=4, fig.width=10}
library(ggplot2)
library(dplyr)
library(viridis)

# Barplot
querylen <- alignrpt_2 %>%
  ggplot( aes(x=QueryLength)) +
    geom_bar(stat="count", fill="green4", binwidth = 50) +
    theme_grey() +
    labs(x = "Query Length", y = "Sequences Count") +
  # xbreaks8 +
  ybreaks10 +
  formatPlot

alignlen <- alignrpt_2 %>%
  ggplot( aes(x=AlignmentLength)) +
    geom_bar(stat="count", fill="green4", binwidth = 50) +
    theme_grey() + xlim(0, 310) +
    labs(x = "Alignment Length", y = "Sequences Count") +
  # xbreaks8 +
  ybreaks10 +    scale_fill_viridis(discrete = TRUE) +
  formatPlot

idperc <- alignrpt_2 %>%
  ggplot( aes(x=PercentIdentity)) +
    geom_bar(stat="count", fill="green4", binwidth = 10) +
    theme_grey() +
    labs(x = "Percent Identity", y = "Sequences Count") +
  # xbreaks8 +
  ybreaks10 +
  formatPlot

ggpubr::ggarrange(querylen, alignlen, idperc, ncol = 3, nrow = 1, align = "hv", labels = "AUTO", font.label = list(size = 12, color = "black", face = "bold", family = NULL))
```

<br>
<br>


## Publication-ready figure
> Subplots of assembled (blue) and aligned (green) sequences are grouped together

```{r contalignpubready}
ggpubr::ggarrange(seqlen, overlaplen, mismatch, querylen, alignlen, idperc, ncol = 3, nrow = 2, align = "hv", labels = "AUTO", font.label = list(size = 12, color = "black", face = "bold", family = NULL))
```



<br>
<br>


## View the first 10 rows of prosecced data
```{bash}
paste -d'\t' \
../../data/mothur/phylotype/00_processed_seqs/qced.contigs.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.good.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count.summary \
>../../data/mothur/phylotype/00_processed_seqs/seqdepth.tsv

echo -e "SampleID\tOriginal\tScreened\tAligned\tDenoised\tNonChimeric\tBacteriaOnly\tNoMock" >../../results/tables/processedSeqs.tsv
cut -f1,2,4,6,8,10,12,14 ../../data/mothur/phylotype/00_processed_seqs/seqdepth.tsv >>../../results/tables/processedSeqs.tsv

```


```{r processedseqdata, message=FALSE, warning=FALSE}
processedseqs <- read_table2(file="../../results/tables/processedSeqs.tsv", col_names = TRUE)
# names(processedseqs)
# dim(processedseqs)

# Define variables (Very very very important!)

processedseqs$SampleID <- as.factor(processedseqs$SampleID)
processedseqs$Original <- as.numeric(processedseqs$Original)
processedseqs$Screened <- as.numeric(processedseqs$Screened)
processedseqs$Aligned <- as.numeric(processedseqs$Aligned)
processedseqs$Denoised <- as.numeric(processedseqs$Denoised)
processedseqs$NonChimeric <- as.numeric(processedseqs$NonChimeric)
processedseqs$BacteriaOnly <- as.numeric(processedseqs$BacteriaOnly)
processedseqs$NoMock <- as.numeric(processedseqs$NoMock)

library(dplyr)
## Summary statistics of processed sequences
processedseqs <- processedseqs %>% dplyr::filter(SampleID != "Mock" & SampleID != "Mock2") # Remove Mock rows. Be sure to remove only the rows that contains the Mock data

processedseqs %>% knitr::kable() %>% kable_styling(bootstrap_options = "bordered", full_width = FALSE)
```

<br>
<br>

## Summary statistics of processed sequences
> Shows summary of the exact number of sequences remaining at each step in 4-quantiles.

```{r}
library(dplyr)
library(kableExtra)
summary(processedseqs[,-1]) %>% knitr::kable() %>% kable_styling(bootstrap_options = "bordered", full_width = FALSE)
```


<br>
<br>


## View the first 10 row of sequence length data

```{bash}
paste -d'\t' \
../../data/mothur/qced.trim.contigs.summary \
../../data/mothur/qced.trim.contigs.good.unique.summary \
../../data/mothur/qced.trim.contigs.good.unique.good.filter.unique.summary \
../../data/mothur/qced.trim.contigs.good.unique.good.filter.unique.precluster.summary \
../../data/mothur/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.summary \
../../data/mothur/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.summary \
../../data/mothur/phylotype/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.summary \
>../../data/mothur/phylotype/00_processed_seqs/seqlenth.tsv

echo -e "Original\tScreened\tAligned\tDenoised\tNonChimeric\tBacteriaOnly\tNoMock" >../../results/tables/seqlenth.tsv
cut -f4,11,18,25,32,39,46 ../../data/mothur/phylotype/00_processed_seqs/seqlenth.tsv >>../../results/tables/seqlenth.tsv

```

```{r}
library(dplyr)
library(readr)
seqlen <- read_table2(file="../../results/tables/seqlenth.tsv", col_names = TRUE)
# names(seqlen)
# dim(seqlen)

# Define variables (Very very very important!)

seqlen$Original <- as.numeric(seqlen$Original)
seqlen$Screened <- as.numeric(seqlen$Screened)
seqlen$Aligned <- as.numeric(seqlen$Aligned)
seqlen$Denoised <- as.numeric(seqlen$Denoised)
seqlen$NonChimeric <- as.numeric(seqlen$NonChimeric)
seqlen$BacteriaOnly <- as.numeric(seqlen$BacteriaOnly)
seqlen$NoMock <- as.numeric(seqlen$NoMock)

## Summary statistics of processed sequences

seqlen[-1,] %>% head(10) %>% knitr::kable() %>% kable_styling(bootstrap_options = "bordered", full_width = FALSE)

```



<br>
<br>

<hr>

## Graphical distribution of processed sequences {-}

<hr>


```{r}
seqcount.v <- cbind(variables, processedseqs)
names(variables)[2] <- "origID"
seqcount.v.m <- reshape2::melt(seqcount.v, id = c("SampleID","Var1","Var2","Var3","Var4"))

# seqcount.v.m <- seqcount.v.m[-c(41:48),]
seqcount.v.m$value <- as.numeric(seqcount.v.m$value)

```

### Stacked barplot {-}
>Note that it for large number of samples it is difficult to plot the x-axis. In such situation it is good to split the samples (see example below)

```{r seqbar1, message=FALSE, warning=FALSE}

library(ggplot2)
library(ggpubr)
library(viridis)

seqcount.v.m$SampleID <- as.factor(seqcount.v.m$SampleID)
seqcount.v.m$Var4 <- as.factor(seqcount.v.m$Var4)
seqcount.v.m$value <- as.numeric(seqcount.v.m$value)
seqcount.v.m$variable <- as.factor(seqcount.v.m$variable)

seqbar1 <- ggplot(seqcount.v.m, aes(x = reorder(SampleID, -value), y = value, fill = variable)) +
geom_bar(stat = "identity", position = "stack") +
labs(x = "Data Points", y = "Sequence depth", title = "", fill = "Dataset") +
theme(axis.text=element_text(size=10)) +
  mediumaxislayout +
  ybreaks10 +
    scale_fill_viridis(discrete = TRUE)
    
ggpubr::ggarrange(seqbar1, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "right", align = "hv", labels = "")

```

<br>

## Subsampling examples
>Below is a command for sampling a specific dataset size. Here it shows how to filter samples with less that 100, 2000, or 5000 sequences. You may want to analyze different subsamples....

<br>

### Samples with less than 100 sequences
```{r echo=TRUE}
library(dplyr)
subsetlt100 <- seqcount.v.m %>% as.data.frame() %>% dplyr::filter(value <100)
subsetlt2000 <- seqcount.v.m %>% as.data.frame() %>% dplyr::filter(value <2000)
subsetlt5000 <- seqcount.v.m %>% as.data.frame() %>% dplyr::filter(value <5000)
```


<br>
<hr>
<br>

### Barplots grouped by process
> Shows maximum sequence depth. 

```{r barforintext1, fig.height=4, fig.width=8}
library(scales)
barforintext1 <- ggplot(seqcount.v.m, aes(x = variable, y = value, fill= variable)) +
  geom_bar(stat = "identity", position = "dodge") +
  ggpubr::color_palette("rainbow")+
  ggpubr::theme_classic2() +
  nolegend + ybreaks10+  xlab("") + ylab("Sequence depth") + ggtitle("") + nolegend +
  theme(axis.text.x=element_text(size=10, angle = 0, hjust = 1)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(axis.title=element_text(size=10,face="bold")) + cleanup + theme(legend.position="none")+
  scale_y_continuous(breaks=pretty_breaks(n=8)) +
  theme(axis.text.x = element_text(hjust=0.5))  +  scale_fill_viridis(discrete = TRUE)

ggpubr::ggarrange(barforintext1, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "top", align = "hv", labels = "")
```

<br>
<br>

### Boxplots grouped by filtering process
```{r, boxforintext2, fig.height=4, fig.width=8}
library(scales)

boxforintext1 <- ggplot(seqcount.v.m, aes(x = variable, y = value, fill = variable)) +
  geom_boxplot(aes(), size = 0.7) +
  geom_jitter(aes(), size = 1, width = 0.20, shape = 19) +
  # ggpubr::color_palette("rainbow")+
  theme_classic() +
  nolegend + ybreaks10 +  labs(y = "Sequence depth", x = "Dataset", title = "", fill = "Dataset") + 
  theme(axis.text.x=element_text(size=10, angle = 0, hjust = 1)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(axis.title=element_text(size=10,face="bold")) + 
  theme(legend.position="none") +
  scale_y_continuous(breaks=pretty_breaks(n=8)) +
  theme(axis.text.x = element_text(hjust=0.5))  +  mediumaxislayout2 + scale_fill_viridis(discrete = TRUE)

ggpubr::ggarrange(boxforintext1, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "right", align = "hv", labels = "")

```

<br>
<br>

### Boxplots grouped by Var1 (Sex in this example)
```{r boxplot2, fig.height=4, fig.width=8}
library(ggpubr)

boxplot2 <- ggboxplot(seqcount.v.m, x = "variable", y = "value",
                color = "Var1", palette =c("rainbow"),
                add = "jitter") +  xlab("") + ylab("Sequence depth") + ggtitle("")  

ggpubr::ggarrange(boxplot2, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "bottom", align = "hv", labels = "")
```

<br>
<br>

### Boxplots grouped by Var2
```{r boxplot3, fig.height=4, fig.width=8}

boxplot3 <- ggboxplot(seqcount.v.m, x = "variable", y = "value",
                color = "Var2", palette =c("rainbow"),
                add = "jitter") +  xlab("") + ylab("Sequence depth") + ggtitle("")  

ggpubr::ggarrange(boxplot3, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "top", align = "hv", labels = "")
```

<br><hr><br>

### Density plots grouped by Var1 and Var2

```{r Var1intext_densplot, fig.height=4, fig.width=8}
Var1intext_densplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack", alpha = 0.8,
   color = "variable", fill = "variable",
   palette = c("rainbow")) + 
    labs(x = "Sequence depth", y ="Density", title = "") +
    nolegend +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=14,face="bold")) +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust=0.5)) +
  facet_grid(~Var1) +
  xbreaks10 +  mediumaxislayout2 + scale_fill_viridis(discrete = TRUE)

Var1intext_densplot

Var2intext_densplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack", alpha = 0.8,
   color = "variable", fill = "variable",
   palette = c("rainbow")) +   
  labs(x = "Sequence depth", y ="Density", title = "") +
    nolegend +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=14,face="bold")) +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust=0.5)) +
  facet_grid(~Var2) +
  xbreaks10+  mediumaxislayout2 + scale_fill_viridis(discrete = TRUE)
Var2intext_densplot
```

<br><hr><br>

### Histograms grouped by Var1 and Var2

```{r Var1intext_histplot, fig.height=4, fig.width=8}
Var1intext_histplot <- gghistogram(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack", alpha = 0.5,
   color = "variable", fill = "variable",
   palette = c("rainbow")) + 
    labs(x = "Sequence depth", y ="Count", title = "") +
    nolegend +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=14,face="bold")) +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust=0.5)) +
  facet_grid(~Var1) +
  xbreaks10
Var1intext_histplot


Var2intext_histplot <- gghistogram(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack", alpha = 0.5,
   color = "variable", fill = "variable",
   palette = c("rainbow")) +   
  labs(x = "Sequence depth", y ="Count", title = "") +
    nolegend +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=12,face="bold")) +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust=0.5)) +
  facet_grid(~Var2) +
  xbreaks10
Var2intext_histplot
```


```{r pubready, fig.height=4, fig.width=8}
### Multiple varieties of plots for same data

seqcount.v.m$variable <- as.factor(seqcount.v.m$variable)
seqcount.v.m$value <- as.numeric(seqcount.v.m$value)


barforintext <- ggplot(seqcount.v.m, aes(x = variable, y = value, fill= variable)) +
  geom_bar(stat = "identity", position = "dodge") +
  ggpubr::color_palette("rainbow")+
  ggpubr::theme_classic2() +
  nolegend + ybreaks10+  xlab("") + ylab("Sequence depth") + ggtitle("") +
  theme(axis.text.x=element_text(size=14, angle = 0, hjust=0.5)) +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=14,face="bold")) + 
  scale_y_continuous(breaks=pretty_breaks(n=8)) +
  theme(legend.position="top") + formatPlot2
# barforintext


boxforintext <- ggplot(seqcount.v.m, aes(x = variable, y = value)) +
  geom_boxplot(aes(color = variable), size = 0.7) +
  geom_jitter(aes(color = variable), size = 1, width = 0.20, shape = 19) +
  ggpubr::color_palette("rainbow")+
  ggpubr::theme_classic2() +
  nolegend + ybreaks10+  xlab("") + ylab("Sequence depth") + ggtitle("") + 
  nolegend +
  theme(axis.text.x=element_text(size=14, angle = 0, hjust = 1)) +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=14,face="bold")) +
    nolegend +
  scale_y_continuous(breaks=pretty_breaks(n=8)) +
  theme(axis.text.x = element_text(hjust=0.5)) + formatPlot2
# boxforintext

Var1intext_densplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack", alpha = 0.5,
   color = "variable", fill = "variable",
   palette = c("rainbow")) + 
    labs(x = "Sequence depth", y ="Density", title = "") +
    nolegend +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=14,face="bold")) +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust=0.5)) +
  facet_grid(~Var1) +
  xbreaks10 + formatPlot2
# Var1intext_densplot

Var2intext_densplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack", alpha = 0.5,
   color = "variable", fill = "variable",
   palette = c("rainbow")) +   
  labs(x = "Sequence depth", y ="Density", title = "") +
    nolegend +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=14,face="bold")) +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust=0.5)) +
  facet_grid(~Var2) +
  xbreaks10 + formatPlot2
# Var2intext_densplot

Var1intext_histplot <- gghistogram(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack", alpha = 0.5,
   color = "variable", fill = "variable",
   palette = c("rainbow")) + 
    labs(x = "Sequence depth", y ="Count", title = "") +
    nolegend +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=14,face="bold")) +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust=0.5)) +
  facet_grid(~Var1) +
  xbreaks10 + formatPlot2
# Var1intext_histplot


Var2intext_histplot <- gghistogram(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack", alpha = 0.5,
   color = "variable", fill = "variable",
   palette = c("rainbow")) +   
  labs(x = "Sequence depth", y ="Count", title = "") +
    nolegend +
  theme(axis.text.y=element_text(size=14)) +
  theme(axis.title=element_text(size=12,face="bold")) +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust=0.5)) +
  facet_grid(~Var2) +
  xbreaks10 + formatPlot2
# Var2intext_histplot
```


<hr>
<br>

## Publication-ready figure

* All subplots on single plot: 
```{r paper_procplots, echo=FALSE, fig.height=14, fig.width=14, message=FALSE, warning=FALSE}

plots <- cowplot::align_plots(barforintext, boxforintext,  Var1intext_densplot, Var1intext_histplot, align = 'v', axis = 'l')
top_row <- cowplot::plot_grid(plots[[1]], labels = "", align = 'h', rel_heights = c(1.4, 1.4))
middle_row1<- cowplot::plot_grid(plots[[2]], labels = "", align = 'h', rel_heights = c(1, 1))
middle_row2 <- cowplot::plot_grid(plots[[3]], Var2intext_densplot, labels = "", align = 'h', rel_widths = c(1, 1), rel_heights = c(1.2, 1.2))
bottom_row <- cowplot::plot_grid(plots[[4]], Var2intext_histplot, labels = "", align = 'h', rel_widths = c(1, 1), rel_heights = c(1.2, 1.2))

ggpubr::ggarrange(top_row, middle_row1,  middle_row2, bottom_row, ncol = 1, nrow=4, common.legend = TRUE, legend = "top", align = "hv", labels = "AUTO")
```

<br>
<hr>
<br>

# Posible questions {-}

<br>

### At sequence processing 
* QN1: Do the paired sequences overlap as expected?
* QN2: Are there sequences to be removed from the analysis based on sequence length?
* QN3: Does the dorminant length match the targeted region of 16S rRNA gene?
* QN4: Are the representative sequences too few or too many? <mark>Note that too many non-redundant sequences may results from poor overlapping between forward and reverse reads. Review before proceeding to avoid misleading conclusions<mark>.


<br>

### At sequence classification {-}
* QN1: What taxonomic classifier should be used? Think of need to develop and use custom classifier that suits the study objectives.
* QN2: Are there sequences assigned to non-bacterial taxonomic lineages?
* QN3: Should the matches to non-bacterial lineages be removed from further analysis?
* QN4: .......?

<br>

### At error estimation {-}
* QN1: Is the error too high to cause rejection of the results results?
* QN2: .......?
* QN3: .......?
* QN4: .......?

<br>
<hr>

## Summary of packages used in the analysis
```{r}
sessionInfo()
```

<br>

