---
title: "Progress report 3"
bibliography: ./library/imap.bib
date: 'Updated: `r Sys.time()`'
output:
  html_document:
    css: ./code/css/report.css
    keep_md: yes
link-citations: yes
subtitle: 'Sequence Processing and classification'
biblio-style: apalike
---


```{r child='00_00_global_setup.Rmd'}

```


<!-- # <mark>Progress report 3</mark> -->

<br>

<!-- # Sequence Processing and onserved Taxonomy Assignment {-} -->

<!-- The clean paired-end sequences that passed the preprocessing trimming and filtering were processed and classified using mothur-based phylotype and OTU-based approaches as described [@Schloss2009]. This step enables investigators to review merged sequences to determine if expected length is in-line with the targeted 16S rRNA gene region. Lack of good overlapping region could results into too many sequences that are wrong and the effect will be propagated in the downstream analysis. High quality sequences align in the same region when searched against a reference template. -->

```{r mergereportdata, eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
contigrpt <- readr::read_table2(file = "./data/mothur/qced.contigs.report")
alignrpt <- readr::read_table2(file = "./data/mothur/qced.trim.contigs.good.unique.align.report")
save(contigrpt, alignrpt, file = "./results/mergealignreport.RData")
```


```{r alignreport, include=FALSE}
load("results//mergealignreport.RData")
contigrpt_2 <- contigrpt[c(2,3,6)]
names(contigrpt_2) <- c("SeqLength","Overlap_Length","MisMatches")

write.table(contigrpt_2,"./results/tables/phylotype/contigrpt_2.tsv", append = FALSE, quote = FALSE, sep="\t", row.names = FALSE, col.names = TRUE, qmethod = c("escape", "double"), fileEncoding = "")


alignrpt_2 <- alignrpt[c(2,12,16)]
names(alignrpt_2) <- c("QueryLength","PairwiseAlignmentLength","SimBtwnQuery.Template")

write.table(alignrpt_2,"./results/tables/phylotype/alignrpt_2.tsv", append = FALSE, quote = FALSE, sep="\t", row.names = FALSE, col.names = TRUE, qmethod = c("escape", "double"), fileEncoding = "")

```

```{r include=FALSE}
p1 <- funModeling::plot_num(contigrpt_2)
p2 <- funModeling::plot_num(alignrpt_2)
```

<br>

## Features of assembled sequences 
```{r fig.height=3, fig.width=8}
p1
```


<br>

## Descriptive statistics of assembled sequence
```{r message=FALSE, warning=FALSE}
knitr::kable(summary(contigrpt_2), "html", position = "center", booktabs = TRUE) %>% kable_styling(bootstrap_options = "striped", font_size = 12)

```

<br>

## Features of aligned sequences 

```{r fig.height=3, fig.width=8}
p2
```

<br>

## Descriptive statistics of aligned sequence

```{r message=FALSE, warning=FALSE}
knitr::kable(summary(alignrpt_2), "html", position = "center", booktabs = TRUE) %>% kable_styling(bootstrap_options = "striped", font_size = 12)

```


```{r processedseqdata, message=FALSE, warning=FALSE}
library(dplyr)
processedseqs <- read_table2(file="./results/tables/processedSeqs.tsv", col_names = TRUE)
# names(processedseqs)
# dim(processedseqs)

processedseqs <- processedseqs[-c(361:362),] # Remove Mock rows

seqcount.v <- cbind(murinevars, processedseqs[,-2])
names(seqcount.v)[7] <- "Group"
# funModeling::df_status(seqcount.v)

seqcount.v.m <- reshape2::melt(seqcount.v, id = c("SampleID", "origGroup", "Sex", "Time", "DayID", "DPW", "Group"))

```

<br>

## Summary statistics of processed sequences
```{r}
# names(seqcount.v[,8:13])
summary(seqcount.v[,8:13])

```

<br>

## Distribution of processed sequences {-}

<hr>

### Stacked barplot {-}

```{r seqbar1, fig.height=5, fig.width=12, message=FALSE, warning=FALSE}

library(ggpubr)

seqbar1 <- ggbarplot(
  seqcount.v.m, x = "SampleID", y = "value",
  fill = "variable",           # change fill color by mpg_level
  color = "variable",            # Set bar border colors to white
  palette = "rainbow",            # jco journal color palett. see ?ggpar# sort.val = "desc",           
  # Sort the value in ascending order
  sort.by.groups = TRUE,     # Don't sort inside each group
  x.text.angle = 0,          # Rotate vertically x axis texts
  ylab = "Sequence depth",
  xlab = "Data Points (360 samples)") + 
  noxlabels

ggpubr::ggarrange(seqbar1, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "right", align = "hv", labels = "")

```

Figure x: Stacked barplots of processed sequences grouped by processing variable

<br>

### Boxplots grouped by processing variable{-}

<br>

```{r boxrpt3_1, fig.height=10, fig.width=10, message=FALSE, warning=FALSE}

library(ggplot2)
library(ggpubr)

seqbar <- ggplot(seqcount.v.m,aes(variable,value, fill=variable)) +
     geom_bar(stat="identity", width=.8, position="dodge")+
  xlab("") + ylab("Sequence depth")  + ggtitle("")+
  scale_y_continuous(breaks=pretty_breaks(n=8)) +
  theme(axis.text.x=element_text(size=10, angle = 0, hjust = 1)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(axis.title=element_text(size=12,face="bold"))+ cleanup+ theme(legend.position="none")


seqpointbox <- ggboxplot(seqcount.v.m, x = "variable", y = "value",
                color = "variable", palette =c("rainbow"),
                add = "point", shape = 19) +  xlab("") + ylab("Sequence depth") + ggtitle("") +
  theme(axis.title=element_text(size=12,face="bold"))+
  theme(axis.text.x=element_text(size=10, angle = 0, hjust = 1)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(axis.title=element_text(size=12,face="bold"))+ cleanup+ theme(legend.position="none")

seqjitterbox <- ggboxplot(seqcount.v.m, x = "variable", y = "value",
                color = "variable", palette =c("rainbow"),
                add = "jitter", shape = 19) +  xlab("") + ylab("Sequence depth") + ggtitle("") +
  theme(axis.title=element_text(size=12,face="bold"))+
  theme(axis.text.x=element_text(size=10, angle = 0, hjust = 1)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(axis.title=element_text(size=12,face="bold"))+ cleanup+ theme(legend.position="none")

ggpubr::ggarrange(seqbar, seqpointbox, seqjitterbox, ncol = 1, nrow = 3,  common.legend = TRUE, legend = "none", align = "hv", labels = "AUTO")
```

<br>

### Boxplots grouped by sex and time variable
```{r boxrpt3_2, fig.height=8, fig.width=10}

boxplot <- ggboxplot(seqcount.v.m, x = "variable", y = "value",
                color = "Time", palette =c("rainbow"),
                add = "jitter", shape = "Time") +  xlab("") + ylab("Sequence depth") + ggtitle("") + facet_grid(~ Sex) + coord_flip()
ggpubr::ggarrange(boxplot, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "top", align = "hv", labels = "")
```


<br>

```{r denshistrpt3, eval=FALSE, fig.height=8, fig.width=10, message=FALSE, warning=FALSE, include=FALSE}
sexdensplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack",
   color = "variable", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Density")+
  theme(axis.text.x=element_text(size=10, angle = 90, hjust = 1)) + facet_grid(~Sex) + ggtitle("")

timedensplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack",
   color = "variable", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Density")+
  theme(axis.text.x=element_text(size=10, angle = 90, hjust = 1)) + facet_grid(~Time) + ggtitle("")

sexhistplot <- gghistogram(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, bins = 25,
   color = "white", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Count")+
  theme(axis.text.x=element_text(size=10, angle = 90, hjust = 1)) + facet_grid(~Sex) + ggtitle("")

timehistplot <- gghistogram(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, bins = 25,
   color = "white", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Count")+
  theme(axis.text.x=element_text(size=10, angle = 90, hjust = 1)) + facet_grid(~Time) + ggtitle("")

# ggpubr::ggarrange(sexdensplot, sexhistplot, timedensplot, timehistplot, ncol = 2, nrow = 2,  common.legend = TRUE, legend = "right", align = "hv", labels = "")
```


<br>

### Publication-ready distribution plots

```{r seqintext_barboxdenshist, fig.height=12, fig.width=12}
seqbar <- ggplot(seqcount.v.m,aes(variable,value, fill=variable)) +
     geom_bar(stat="identity", width=.8, position="dodge")+
  xlab("") + ylab("Sequence depth")  + ggtitle("")+
  scale_y_continuous(breaks=pretty_breaks(n=8)) +
  theme(axis.text.x=element_text(size=12, angle = 0, hjust = 1)) +
  theme(axis.text.y=element_text(size=12)) +
  theme(axis.title=element_text(size=14,face="bold"))+ cleanup+ theme(legend.position="top")

boxforintext2 <- ggboxplot(seqcount.v.m, x = "variable", y = "value",
                color = "variable", palette =c("rainbow"),
                add = "jitter", shape = 19) +  xlab("") + ylab("Sequence depth") + ggtitle("") +
  theme(axis.text.x=element_text(size=12, angle = 90, hjust = 1)) + nolegend +
  theme(axis.text.x=element_text(size=12, angle = 0, hjust = 1)) +
  theme(axis.text.y=element_text(size=12)) +
  theme(axis.title=element_text(size=14,face="bold"))+ cleanup+ theme(legend.position="none")+
  scale_y_continuous(breaks=pretty_breaks(n=8))

sexintext_densplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack",
   color = "variable", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Density") +
  theme(axis.text.x=element_text(size=12, angle = 90, hjust = 1)) + facet_grid(~Sex) + nolegend +
  theme(axis.text.x=element_text(size=12, angle = 90, hjust = 1)) +
  theme(axis.text.y=element_text(size=12)) +
  theme(axis.title=element_text(size=14,face="bold"))+ cleanup+ theme(legend.position="none")+
  scale_x_continuous(breaks=pretty_breaks(n=8))


sexintext_histplot <- gghistogram(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, bins = 25,
   color = "white", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Count")+
  theme(axis.text.x=element_text(size=12, angle = 90, hjust = 1)) + facet_grid(~Sex)  + nolegend +
  theme(axis.text.x=element_text(size=12, angle = 90, hjust = 1)) +
  theme(axis.text.y=element_text(size=12)) +
  theme(axis.title=element_text(size=14,face="bold"))+ cleanup+ theme(legend.position="none")+
  scale_x_continuous(breaks=pretty_breaks(n=8))

timeintext_densplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack",
   color = "variable", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Density") +
  theme(axis.text.x=element_text(size=12, angle = 90, hjust = 1)) + facet_grid(~Time) + nolegend+
  theme(axis.text.x=element_text(size=12, angle = 90, hjust = 1)) +
  theme(axis.text.y=element_text(size=12)) +
  theme(axis.title=element_text(size=14,face="bold"))+ cleanup+ theme(legend.position="none")+
  scale_x_continuous(breaks=pretty_breaks(n=8))


timeintext_histplot <- gghistogram(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, bins = 25,
   color = "white", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Count")+
  theme(axis.text.x=element_text(size=12, angle = 90, hjust = 1)) + facet_grid(~Time) + nolegend +
  theme(axis.text.x=element_text(size=12, angle = 90, hjust = 1)) +
  theme(axis.text.y=element_text(size=12)) +
  theme(axis.title=element_text(size=14,face="bold"))+ cleanup+ theme(legend.position="none")+
  scale_x_continuous(breaks=pretty_breaks(n=8))


plots <- cowplot::align_plots(boxforintext2, sexintext_densplot,  sexintext_histplot, align = 'v', axis = 'l')
top_row <- cowplot::plot_grid(plots[[1]], labels = "", align = 'h')
middle_row <- cowplot::plot_grid(plots[[2]], timeintext_densplot, labels = "", align = 'h', rel_widths = c(1, 1))
bottom_row <- cowplot::plot_grid(plots[[3]], timeintext_histplot, labels = "", align = 'h', rel_widths = c(1, 1))

ggpubr::ggarrange(seqbar, top_row,middle_row, bottom_row, ncol = 1, nrow = 4)
```

<br>

<br>

# Posible questions {-}

<hr>

### At sequence processing 
* QN1: Do the paired sequences overlap as expected?
* QN2: Are there sequences to be removed from the analysis based on sequence length?
* QN3: Does the dorminant length match the targeted region of 16S rRNA gene?
* QN4: Are the representative sequences too few or too many? <mark>Note that too many non-redundant sequences may results from poor overlapping between forward and reverse reads. Review before proceeding to avoid misleading conclusions<mark>.


<br>

### At sequence classification {-}
* QN1: What taxonomic classifier should be used? Think of need to develop and use custom classifier that suits the study objectives.
* QN2: Are there sequences assigned to non-bacterial taxonomic lineages?
* QN3: Should the matches to non-bacterial lineages be removed from further analysis?
* QN4: .......?

<br>

### At error estimation {-}
* QN1: Is the error too high to cause rejection of the results results?
* QN2: .......?
* QN3: .......?
* QN4: .......?

<br>

<!-- ### At OTU clustering and conserved taxonomy assignment {-} -->
<!-- * QN1: What cutoff should be used? -->
<!-- * QN2: .......? -->
<!-- * QN3: .......? -->
<!-- * QN4: .......? -->

<br>

```{bash eval=FALSE, include=FALSE}
# Bash command

# Rscript -e "rmarkdown::render('./report3.Rmd', clean=TRUE, quiet=FALSE, output_file='report3_sequence_processing.html', output_dir='reports')"

```

<br>

