---
title: "Progress report 4"
bibliography: ./library/imap.bib
date: 'Updated: `r Sys.time()`'
output:
  html_document:
    css: ./code/css/report.css
    keep_md: yes
link-citations: yes
subtitle: 'OTU clustering, Taxonomy assignment & Preliminary analysis
'
biblio-style: apalike
---


```{r child='00_00_global_setup.Rmd'}

```

```{r}
load("./results/permethodtaxdata.RData")
```

<br>

## Conserved taxonomy assignment

### Methods:

* Phylotype method at genus level.
* OTU-based method with minimum set at 97% identity. The opticlust algorithm was used to optimize the quality of assigned taxonomy.
* Phylogeny method where tree was taxonomically classified with minimum set at 97% identity


```{r taxonomydata, eval=FALSE, warning=FALSE, , message=FALSE, include=FALSE}

## Phylotype approach

# For raw abundance analysis
phylotype_shared <- readr::read_table2(file = "data/mothur/phylotype/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.seed.wang.pick.pick.tx.shared")
phylotype_shared.v <- cbind(murinevars, phylotype_shared)

# For rank abundance analysis
phylotyperabund <- readr::read_table2(file = "data/mothur/phylotype/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.seed.wang.pick.pick.tx.rabund")
dim(phylotyperabund)

# For taxon analysis
phylotypetaxsummary <- readr::read_table2(file = "data/mothur/phylotype/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.seed.wang.pick.pick.tx.1.cons.tax.summary")
dim(phylotypetaxsummary)

# For taxonomy Venn diagram
phylotypeconstaxonomy <- readr::read_table2(file = "data/mothur/phylotype/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.seed.wang.pick.pick.tx.1.cons.taxonomy", col_names = TRUE)
dim(phylotypeconstaxonomy)

# For relative abundance analysis
phylotypetaxlefse <- readr::read_table2(file = "data/mothur/phylotype/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.seed.wang.pick.pick.tx.1.lefse")
dim(phylotypetaxlefse)

# For raw abundance analysis
phylotypetaxbiom <- readr::read_table2(file = "data/mothur/phylotype/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.seed.wang.pick.pick.tx.1.biom")
dim(phylotypetaxbiom)

#---------------------
dir.create("results/tables/phylotype")
write.table(phylotypeconstaxonomy,"./results/tables/phylotype/phylotypeconstaxonomy.tsv", append = FALSE, quote = FALSE, sep="\t", row.names = FALSE, col.names = TRUE, qmethod = c("escape", "double"), fileEncoding = "")
#---------------------


## OTUbased approach
# For taxon analysis
otubasedtaxsummary <- readr::read_table2(file = "data/mothur/otus/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.unique_list.0.03.cons.tax.summary")
dim(otubasedtaxsummary)

# For taxonomy Venn diagram
otubasedconstaxonomy <- readr::read_table2(file = "data/mothur/otus/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.unique_list.0.03.cons.taxonomy", col_names = TRUE)
dim(otubasedconstaxonomy)

# For raw abundance analysis
otubased_shared <- readr::read_table2(file = "data/mothur/otus/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.unique_list.shared")
dim(otubased_shared)

otubased_shared_2 <- cbind(murinevars, otubased_shared)
dim(otubased_shared_2)

# For significant statistics
otubasedmccsensspec <- readr::read_table2(file = "data/mothur/otus/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.unique_list.sensspec")
dim(otubasedmccsensspec)

# For relative abundance analysis
otubasedtaxlefse <- readr::read_table2(file = "data/mothur/otus/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.unique_list.0.03.lefse")
dim(otubasedtaxlefse)

# For raw abundance analysis
otubasedtaxbiom <- readr::read_table2(file = "data/mothur/otus/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.unique_list.0.03.biom")

dim(otubasedtaxbiom)


#---------------------
dir.create("results/tables/otus.0.03")
write.table(otubasedconstaxonomy,"./results/tables/otus.0.03/otubasedconstaxonomy.tsv", append = FALSE, quote = FALSE, sep="\t", row.names = FALSE, col.names = TRUE, qmethod = c("escape", "double"), fileEncoding = "")
#---------------------


## Phylogeny approach
# For species richness

phylogenydiversity <- readr::read_table2(file = "data/mothur/phylogeny/01_rarefy/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.phylip.1.phylodiv.summary", col_names = TRUE)
dim(phylogenydiversity)

# For taxonomy Venn diagram
phylogenyconstaxonomy <- readr::read_table2(file = "data/mothur/phylogeny/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.phylip.taxonomy.summary", col_names = TRUE)
dim(phylogenyconstaxonomy)

#---------------------
dir.create("results/tables/phylogeny")
write.table(phylogenyconstaxonomy,"./results/tables/phylogeny/phylogenyconstaxonomy.tsv", append = FALSE, quote = FALSE, sep="\t", row.names = FALSE, col.names = TRUE, qmethod = c("escape", "double"), fileEncoding = "")

# Create iMAP objects for future use
save(otubased_shared, otubased_shared_2, otubasedconstaxonomy, otubasedmccsensspec, otubasedtaxbiom, otubasedtaxlefse, otubasedtaxsummary, phylogenyconstaxonomy, phylogenydiversity, phylotype_shared, phylotype_shared.v, phylotypeconstaxonomy, phylotyperabund, phylotypetaxlefse, phylotypetaxbiom, phylotypetaxsummary, file = "./results/permethodtaxdata.RData")

```



<br>

### Quality of conserved taxonomy
The opticlust algorithm yielded high quality OTUs based on the high precision metrics below.

<br>

|	Parameter	|	Value	|
| :---	| --|	
|	cutoff	|	0.2	|
|	Sensitivity	|	0.998	|
|	Specificity	|	0.999	|
|	PPV	|	0.998	|
|	NPV	|	0.999	|
|	Accuracy	|	0.999	|
|	MCC	|	0.997	|
|	F1score	|	0.998	|
|	FDR	|	0.002	|

Table: Table x: Statistical parameters calculated from the OTU-based classification method

```{r}
load(file = "./results/origotutaxonomy.RData")

```


<!-- <br> -->

<!-- ## Most frequent taxonomy terms -->

```{r eval=FALSE, child='wordcloudtaxon.Rmd', include=FALSE}

```


```{r taxoncloud, eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
set.seed(110912)
library(tm)

# taxon4wordcloud <- readr::read_table2(file = "results/tables/wordcloud/merged.tsv")
taxon4wordcloud <- readr::read_table2(file = "results/tables/wordcloud/taxmerged4wordcloud.txt")

phylumwords <- taxon4wordcloud$Phylum
write.table(phylumwords,"./results/tables/wordcloud/phylumwords.tsv",sep="\t", row.names = FALSE, col.names = FALSE)

# Load the data as a corpus
docs <- Corpus(VectorSource(phylumwords))

# inspect(docs)

toSpace <- content_transformer(function (x , pattern ) gsub(pattern, " ", x))
docs <- tm_map(docs, toSpace, "/")
docs <- tm_map(docs, toSpace, "@")
docs <- tm_map(docs, toSpace, "\\|")

# Convert the text to lower case
docs <- tm_map(docs, content_transformer(tolower))
docs <- tm_map(docs, removeWords, stopwords("english"))
# Remove punctuations
docs <- tm_map(docs, removePunctuation)
# Eliminate extra white spaces
docs <- tm_map(docs, stripWhitespace)
# Text stemming
# docs <- tm_map(docs, stemDocument)
	tdm <- TermDocumentMatrix(docs)
	m <- as.matrix(tdm)
	v <- sort(rowSums(m),decreasing=TRUE)
	p_worddf <- data.frame(word = names(v),freq=v)
	p_worddf[2] <- log2(p_worddf[2])

#####################

classwords <- taxon4wordcloud$Class
write.table(classwords,"./results/tables/wordcloud/classwords.tsv",sep="\t", row.names = FALSE, col.names = FALSE)
# Load the data as a corpus
docs <- Corpus(VectorSource(classwords))

# inspect(docs)

toSpace <- content_transformer(function (x , pattern ) gsub(pattern, " ", x))
docs <- tm_map(docs, toSpace, "/")
docs <- tm_map(docs, toSpace, "@")
docs <- tm_map(docs, toSpace, "\\|")

# Convert the text to lower case
docs <- tm_map(docs, content_transformer(tolower))
docs <- tm_map(docs, removeWords, stopwords("english"))
# Remove punctuations
docs <- tm_map(docs, removePunctuation)
# Eliminate extra white spaces
docs <- tm_map(docs, stripWhitespace)
# Text stemming
# docs <- tm_map(docs, stemDocument)
	tdm <- TermDocumentMatrix(docs)
	m <- as.matrix(tdm)
	v <- sort(rowSums(m),decreasing=TRUE)
	c_worddf <- data.frame(word = names(v),freq=v)
	c_worddf[2] <- log2(c_worddf[2])

#####################

orderwords <- taxon4wordcloud$Order
write.table(orderwords,"./results/tables/wordcloud/orderwords.tsv",sep="\t", row.names = FALSE, col.names = FALSE)

# Load the data as a corpus
docs <- Corpus(VectorSource(orderwords))

# inspect(docs)

toSpace <- content_transformer(function (x , pattern ) gsub(pattern, " ", x))
docs <- tm_map(docs, toSpace, "/")
docs <- tm_map(docs, toSpace, "@")
docs <- tm_map(docs, toSpace, "\\|")

# Convert the text to lower case
docs <- tm_map(docs, content_transformer(tolower))
docs <- tm_map(docs, removeWords, stopwords("english"))
# Remove punctuations
docs <- tm_map(docs, removePunctuation)
# Eliminate extra white spaces
docs <- tm_map(docs, stripWhitespace)
# Text stemming
# docs <- tm_map(docs, stemDocument)
	tdm <- TermDocumentMatrix(docs)
	m <- as.matrix(tdm)
	v <- sort(rowSums(m),decreasing=TRUE)
	o_worddf <- data.frame(word = names(v),freq=v)
	o_worddf[2] <- log2(o_worddf[2])

#####################

familywords <- taxon4wordcloud$Family
write.table(familywords,"./results/tables/wordcloud/familywords.tsv",sep="\t", row.names = FALSE, col.names = FALSE)

# Load the data as a corpus
docs <- Corpus(VectorSource(familywords))

# inspect(docs)

toSpace <- content_transformer(function (x , pattern ) gsub(pattern, " ", x))
docs <- tm_map(docs, toSpace, "/")
docs <- tm_map(docs, toSpace, "@")
docs <- tm_map(docs, toSpace, "\\|")

# Convert the text to lower case
docs <- tm_map(docs, content_transformer(tolower))
docs <- tm_map(docs, removeWords, stopwords("english"))
# Remove punctuations
docs <- tm_map(docs, removePunctuation)
# Eliminate extra white spaces
docs <- tm_map(docs, stripWhitespace)
# Text stemming
# docs <- tm_map(docs, stemDocument)
	tdm <- TermDocumentMatrix(docs)
	m <- as.matrix(tdm)
	v <- sort(rowSums(m),decreasing=TRUE)
	f_worddf <- data.frame(word = names(v),freq=v)
	f_worddf[2] <- log2(f_worddf[2])

#####################

genuswords <- taxon4wordcloud$Genus
write.table(genuswords,"./results/tables/wordcloud/genuswords.tsv",sep="\t", row.names = FALSE, col.names = FALSE)
# Load the data as a corpus
docs <- Corpus(VectorSource(genuswords))

# inspect(docs)

toSpace <- content_transformer(function (x , pattern ) gsub(pattern, " ", x))
docs <- tm_map(docs, toSpace, "/")
docs <- tm_map(docs, toSpace, "@")
docs <- tm_map(docs, toSpace, "\\|")

# Convert the text to lower case
docs <- tm_map(docs, content_transformer(tolower))
docs <- tm_map(docs, removeWords, stopwords("english"))
# Remove punctuations
docs <- tm_map(docs, removePunctuation)
# Eliminate extra white spaces
docs <- tm_map(docs, stripWhitespace)
# Text stemming
# docs <- tm_map(docs, stemDocument)
tdm <- TermDocumentMatrix(docs)
m <- as.matrix(tdm)
v <- sort(rowSums(m),decreasing=TRUE)
g_worddf <- data.frame(word = names(v),freq=v)
g_worddf[2] <- log2(g_worddf[2])
```


```{r wordcloud2, eval=FALSE, include=FALSE}
## Create taxon wordcloud

p_cloud <- wordcloud2(p_worddf, size = .15, rotateRatio = 0, shape = 'square', color = ifelse(p_worddf[, 2] >10, '#cc9900', 'magenta'), backgroundColor = "white")
p_cloud

c_cloud <- wordcloud2(c_worddf, size = .15, rotateRatio = 0, shape = 'square', color = ifelse(p_worddf[, 2] >10, '#cc9900', 'magenta'), backgroundColor = "white")
c_cloud

o_cloud <- wordcloud2(o_worddf, size = .15, rotateRatio = 0, shape = 'square', color = ifelse(p_worddf[, 2] >10, '#cc9900', 'magenta'), backgroundColor = "white")
o_cloud

f_cloud <- wordcloud2(f_worddf, size = .15, rotateRatio = 0, shape = 'square', color = ifelse(p_worddf[, 2] >10, '#cc9900', 'magenta'), backgroundColor = "white")
f_cloud

g_cloud <- wordcloud2(g_worddf, size = .15, rotateRatio = 0, shape = 'square', color = ifelse(p_worddf[, 2] >10, '#cc9900', 'magenta'), backgroundColor = "white")
g_cloud

```

<!-- <br> -->

```{r eval=FALSE, include=FALSE}
# save it in html & png
saveWidget(p_cloud,"p_cloud.html",selfcontained = F)
webshot("p_cloud.html","p_cloud.png", delay =5, vwidth = 480, vheight=480) # changed to png. 

saveWidget(c_cloud,"c_cloud.html",selfcontained = F)
webshot("c_cloud.html","c_cloud.png", delay =5, vwidth = 480, vheight=480) # changed to png. 

saveWidget(o_cloud,"o_cloud.html",selfcontained = F)
webshot("o_cloud.html","o_cloud.png", delay =5, vwidth = 480, vheight=480) # changed to png. 

saveWidget(f_cloud,"f_cloud.html",selfcontained = F)
webshot("f_cloud.html","f_cloud.png", delay =5, vwidth = 480, vheight=480) # changed to png. 

saveWidget(g_cloud,"g_cloud.html",selfcontained = F)
webshot("g_cloud.html","g_cloud.png", delay =5, vwidth = 480, vheight=480) # changed to png. 

```



## OTU abundance {-}

<hr>

* Integrating output obtained from diverse methods could be a robust way for profiling complex microbial communities present in a given sample.
* The phylotype-based approach yielded 197 OTUs at genus level
* The OTU-based approach generated 11,257 OTU clusters at 97% identity. 
* The phylogeny approach generated 58,929 tree nodes which were taxonomically classified at 80% identity. 
* Manual review and annotation should lead to actual non-redundant results.

<br>


![](img/otuabund.png)

Figure x: Unfiltered and curated OTU abundance. Visual representation of taxon terms highlight the most abundant taxon based on frequency of being assigned to an OTU or tree nodes. Muribaculaceae is the most frequently assigned family and Muribaculaceae_ge is the most frequent species assigned to most sequences. 


<br>

### Most abundant species (top 15%)

<hr>

```{r fig.height=6, fig.width=10, message=FALSE, warning=FALSE}
## Calculate DCA
     library(vegan)
     library(goeveg)

     otu.dca <- decorana(otuprop[,-1])
     p.dca <- decorana(phylumprop[,-1])
     c.dca <- decorana(classprop[,-1])
     o.dca <- decorana(orderprop[,-1])
     f.dca <- decorana(familyprop[,-1])
    g.dca <- decorana(genusprop[,-1])
     
     
     ## Select the 15% most abundant species and call the result     
     otu.abund15 <- ordiselect(otuprop[,-1], otu.dca, ablim = 0.15)
     p.abund15 <- ordiselect(phylumprop[,-1], p.dca, ablim = 0.15)
     c.abund15 <- ordiselect(classprop[,-1], c.dca, ablim = 0.15)
     o.abund15 <- ordiselect(orderprop[,-1], o.dca, ablim = 0.15)
     f.abund15 <- ordiselect(familyprop[,-1], f.dca, ablim = 0.15)
     g.abund15 <- ordiselect(genusprop[,-1], g.dca, ablim = 0.15)


cat("\nOTUs\n")
 
otu.abund15

cat("\nPhylum\n")

p.abund15

cat("\nClass\n")

c.abund15

cat("\nOrder\n")

o.abund15

cat("\nFamily\n")

f.abund15

cat("\nGenus\n")

g.abund15

```


<br>

### Example of rank abundance

<hr>

* Demonstration using four samples
* Rank abundance curves can be used to display species richness and species evenness across selected samples.

```{r rankabund, fig.height=6, fig.width=10, message=FALSE, warning=FALSE}
library(goeveg)

## Draw multiple rank-abundance curves for selected samples with coloured lines

par(mfrow = c(2,3))
cex <- par(cex.axis=1.2, cex.lab=1.4, cex.main=1.4, cex.sub=.8)

otu.rank <-racurves(otuprop[c(1,166,174,287),-1], bw = FALSE, main = "OTUs") + normalaxislayout
p.rank <- racurves(phylumprop[c(1,166,174,287),-1], bw = FALSE, main = "Phylum" ) + normalaxislayout
c.rank <- racurves(classprop[c(1,166,174,287),-1], bw = FALSE, main = "Class" ) + normalaxislayout
o.rank <- racurves(orderprop[c(1,166,174,287),-1], bw = FALSE, main = "Order" ) + normalaxislayout
f.rank <- racurves(familyprop[c(1,166,174,287),-1], bw = FALSE, main = "Family" ) + normalaxislayout
g.rank <- racurves(genusprop[c(1,166,174,287),-1], bw = FALSE, main = "Genus" ) + normalaxislayout

```

Figure x. Rank abundance of eight selected samples. Package: [goeveg](https://rdrr.io/cran/goeveg/man/racurves.html)

<br>


<br>

### Correlation between observed phyla

<hr>

<!-- * Seven different visualization methods can be used: -->
<!--   * circle -->
<!--   * square -->
<!--   * ellipse -->
<!--   * number -->
<!--   * shade -->
<!--   * color -->
<!--   * pie -->
<!--   * mixed -->
<!--   * R package [corrplot](https://cran.r-project.org/web/packages/corrplot/corrplot.pdf). -->

```{r correlationdata, fig.height=10, fig.width=10, message=FALSE, warning=FALSE}
otu_corr<-cor(otuprop[,-1])
p_corr<-cor(phylumprop[,-1])
c_corr<-cor(classprop[,-1])
o_corr<-cor(orderprop[,-1])
f_corr<-cor(familyprop[,-1])
g_corr<-cor(genusprop[,-1])


```


<br>

```{r corrcircle, fig.height=12, fig.width=12, message=FALSE, warning=FALSE}
par(mfrow =c(2,2))
par(cex.axis=1, cex.lab=1.5, cex.main=2, cex.sub=1)
corrplot::corrplot(p_corr, method = "circle", diag = TRUE, order = "alphabet", tl.cex = 1)

corrplot::corrplot(p_corr, method = "color", diag = TRUE, order = "alphabet",  tl.cex = 1)

corrplot::corrplot(p_corr, method = "circle", diag = TRUE, order = "hclust",  tl.cex = 1)

corrplot::corrplot(p_corr, method = "color", diag = TRUE, order = "hclust",  tl.cex = 1)
```

Figure x. Correlation between species identified at phylum-level. Species are ordered alphabetically (top panel) and heuristically (bottom panel)


<br>

## Alpha diversity {-}

<hr>

```{r alpharichdiv, fig.height=4, fig.width=8, message=FALSE, warning=FALSE}
mergedrichdiv <- readr::read_table2(file = "results/tables/merged/merged_richness_diversity.tsv")

library(scales)

spprich.m <- reshape2::melt(mergedrichdiv[,c(1:6,11,15)], id=c("SampleID", "OrigID", "Sex", "Time", "DayID", "DPW"))
sppddiv.m <- reshape2::melt(mergedrichdiv[,c(1:6,8,12,18)], id=c("SampleID", "OrigID", "Sex", "Time", "DayID", "DPW"))

```
 
<br>

### Species richness at each datapoint
```{r spprichbar, fig.height=4, fig.width=10, message=FALSE, warning=FALSE}
library(ggpubr)

spprichbar1 <- ggbarplot(spprich.m, x = "SampleID", y = "value",
          fill = "variable",           # change fill color by variable
          color = "variable",            # Set bar border colors to white
          palette = c("rainbow"),            # jco journal color palett. see ?ggpar
          # palette = c("#00AFBB", "#E7B800"),            # jco journal color palett. see ?ggpar
          # sort.val = "desc",           # Sort the value in ascending order
          sort.by.groups = TRUE,     # Don't sort inside each group
          x.text.angle = 0,          # Rotate vertically x axis texts
          ylab = "Species richness",
          xlab = "Data Points (360 samples)", 
          legend.title = "Calculator"
          ) + noxlabels
ggpubr::ggarrange(spprichbar1, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "right", align = "hv", labels = "")

```

Figure x: Stacked barplots for species richness. The estimated richness (green bars) was calculated using **chao** calculator and observed ichness (red bars) was calculated using **sobs**.

<br>

### Species richness: Boxplots, density plots and histograms

```{r boxspprich, fig.height=10, fig.width=10}

boxspprich <- ggboxplot(spprich.m, x = "Time", y = "value",
                color = "variable",  palette =c("#00AFBB", "#E7B800"),
                add = "point", shape = 19, 
          legend.title = "Calculator") +  xlab("Time period") + ylab("Species richness") + ggtitle("") + facet_grid(~ Sex) 

densspprich <- ggdensity(spprich.m, x = "value",
                color = "variable",  palette =c("#00AFBB", "#E7B800"),
                add = "mean", shape = 19, 
          legend.title = "Calculator") +  xlab("Species richness") + ylab("Density") + ggtitle("") + facet_grid(~ Sex) 

histspprich <- gghistogram(spprich.m, x = "value",
                color = "white", fill = "variable", palette =c("#00AFBB", "#E7B800"),
                add = "mean", shape = 19, 
          legend.title = "Calculator") +  xlab("Species richness") + ylab("Frequency") + ggtitle("") + facet_grid(~ Sex) 

ggpubr::ggarrange(boxspprich, densspprich, histspprich, ncol = 1, nrow = 3,  common.legend = TRUE, legend = "right", align = "hv", labels = "AUTO")

```

Figure x: Species richness (observed species) displayed by boxplot (A), density plots (B) and histograms (C).

<br>

### Correlation between species richness and sequence depth

```{r corrspprich, fig.height=7, fig.width=8}

sobsvsseqdepth <- ggscatter(
  mergedrichdiv, x = "sobs", y = "numSampled", color = "Time",  palette =c("#00AFBB", "#E7B800"), add = "none", shape = 19) + 
  xlab("Observed species richness") + 
  ylab("Sequence Depth") + 
  ggtitle("") + 
  facet_grid(~ Sex) + 
  stat_cor(method = "spearman") + normalaxislayout

chaovsseqdepth <- ggscatter(
  mergedrichdiv, x = "chao", y = "numSampled", color = "Time",  palette =c("#00AFBB", "#E7B800"), add = "none", shape = 19) +
  xlab("Estimated species richness") +
  ylab("Sequence Depth") +
  ggtitle("") +
  facet_grid(~ Sex) +
  stat_cor(method = "spearman") + normalaxislayout

ggpubr::ggarrange(sobsvsseqdepth, chaovsseqdepth, ncol = 1, nrow = 2,  common.legend = FALSE, legend = "right", align = "hv", labels = "AUTO")

```


Figure x: Correlation between species richness and sequence depth. Observed species calculated using **sobs** (A) and estimated species richness by **chao** calculator (B).

<br>


### Correlation between species diversity and species richness

<br>

```{r sppdiversity, fig.height=10, fig.width=8}
invsppdiv <- ggline(
  mergedrichdiv, x = "sobs", y = "invsimpson", color = "Time",  palette =c("#00AFBB", "#E7B800"), add = "none", shape = 19) + 
  xlab("Observed species richness") + 
  ylab("Simpson-Diversity") + 
  ggtitle("") + 
  facet_grid(~ Sex) + 
  stat_cor(method = "spearman") + normalaxislayout + noxtitle

shansppdiv <- ggline(
  mergedrichdiv, x = "sobs", y = "npshannon", color = "Time",  palette =c("#00AFBB", "#E7B800"), add = "none", shape = 19) + 
  xlab("Observed species richness") + 
  ylab("Shannon-Diversity") + 
  ggtitle("") + facet_grid(~ Sex) + 
  stat_cor(method = "spearman") + normalaxislayout + noxtitle

phylopdiv <- ggline(
  mergedrichdiv, x = "sobs", y = "phyloDiversity", color = "Time",  palette =c("#00AFBB", "#E7B800"), add = "none", shape = 19) + 
  xlab("Observed species richness") + 
  ylab("Phylo-Diversity") + 
  ggtitle("") + 
  facet_grid(~ Sex) + 
  stat_cor(method = "spearman") + normalaxislayout


ggpubr::ggarrange(invsppdiv, shansppdiv, phylopdiv, ncol = 1, nrow = 3,  common.legend = FALSE, legend = "right", align = "hv", labels = "AUTO")
```


```{r spprichdiversity, fig.height=14, fig.width=8, include=FALSE}
ggpubr::ggarrange(sobsvsseqdepth, chaovsseqdepth,invsppdiv, shansppdiv, phylopdiv, ncol = 1, nrow = 5,  common.legend = FALSE, legend = "right", align = "hv", labels = "AUTO")
```


Figure x: Species diversity and correlation to species richness. Definitely phylo-diversity (C) correlates well with the species richness.


```{r corrboxdenshist, eval=FALSE, fig.height=14, fig.width=10, message=FALSE, warning=FALSE, include=FALSE}
ggpubr::ggarrange(corrsobsseqdepth, invsppdiv, shansppdiv, phylopdiv, ncol = 1, nrow = 4,  common.legend = TRUE, legend = "top", align = "hv", labels = "AUTO")
```

<br>

## Species accumulation

<hr>

* Methods used
  * Exact
  * Random
  * Collector
  * Rarefaction


```{r loadtaxadata, eval=FALSE, include=FALSE}
# all taxa data
phylum.a <- read.table(file = "results/tables/phylotype/phylum.a.tsv", header = TRUE, sep = "\t")
class.a <- read.table(file = "results/tables/phylotype/class.a.tsv", header = TRUE, sep = "\t")
order.a <- read.table(file = "results/tables/phylotype/order.a.tsv", header = TRUE, sep = "\t")
family.a <- read.table(file = "results/tables/phylotype/family.a.tsv", header = TRUE, sep = "\t")
genus.a <- read.table(file = "results/tables/phylotype/genus.a.tsv", header = TRUE, sep = "\t")

# taxonfiltered
otu <- read.table(file = "results/tables/phylotype/otu.tsv", header = TRUE, sep = "\t")
phylum <- read.table(file = "results/tables/phylotype/phylum.tsv", header = TRUE, sep = "\t")
class <- read.table(file = "results/tables/phylotype/class.tsv", header = TRUE, sep = "\t")
order <- read.table(file = "results/tables/phylotype/order.tsv", header = TRUE, sep = "\t")
family <- read.table(file = "results/tables/phylotype/family.tsv", header = TRUE, sep = "\t")
genus <- read.table(file = "results/tables/phylotype/genus.tsv", header = TRUE, sep = "\t")

# transposed
otu.t <- read.table(file = "results/tables/phylotype/otu.t.tsv", header = TRUE, sep = "\t")
phylum.t<- read.table(file = "results/tables/phylotype/phylum.t.tsv", header = TRUE, sep = "\t")
class.t <- read.table(file = "results/tables/phylotype/class.t.tsv", header = TRUE, sep = "\t")
order.t <- read.table(file = "results/tables/phylotype/order.t.tsv", header = TRUE, sep = "\t")
family.t <- read.table(file = "results/tables/phylotype/family.t.tsv", header = TRUE, sep = "\t")
genus.t <- read.table(file = "results/tables/phylotype/genus.t.tsv", header = TRUE, sep = "\t")

rownames(otu.t) <- otu.t[, 1] # Set the rownames form phylum
rownames(phylum.t) <- phylum.t[, 1] # Set the rownames
rownames(class.t) <- class.t[, 1] # Set the rownames
rownames(order.t) <- order.t[, 1] # Set the rownames
rownames(family.t) <- family.t[, 1] # Set the rownames
rownames(genus.t) <- genus.t[, 1] # Set the rownames

# with variables
otu.t.v <- read.table(file = "results/tables/phylotype/otu.t.v.tsv", header = TRUE, sep = "\t")
phylum.t.v<- read.table(file = "results/tables/phylotype/phylum.t.v.tsv", header = TRUE, sep = "\t")
class.t.v <- read.table(file = "results/tables/phylotype/class.t.v.tsv", header = TRUE, sep = "\t")
order.t.v <- read.table(file = "results/tables/phylotype/order.t.v.tsv", header = TRUE, sep = "\t")
family.t.v <- read.table(file = "results/tables/phylotype/family.t.v.tsv", header = TRUE, sep = "\t")
genus.t.v <- read.table(file = "results/tables/phylotype/genus.t.v.tsv", header = TRUE, sep = "\t")

# proportions
otuprop <- read.table(file = "results/tables/phylotype/otuprop.tsv", header = TRUE, sep = "\t")
phylumprop<- read.table(file = "results/tables/phylotype/phylumprop.tsv", header = TRUE, sep = "\t")
classprop <- read.table(file = "results/tables/phylotype/classprop.tsv", header = TRUE, sep = "\t")
orderprop <- read.table(file = "results/tables/phylotype/orderprop.tsv", header = TRUE, sep = "\t")
familyprop <- read.table(file = "results/tables/phylotype/familyprop.tsv", header = TRUE, sep = "\t")
genusprop <- read.table(file = "results/tables/phylotype/genusprop.tsv", header = TRUE, sep = "\t")

rownames(otuprop) <- otu.t[, 1] # Set the rownames form phylum
rownames(phylumprop) <- phylum.t[, 1] # Set the rownames
rownames(classprop) <- class.t[, 1] # Set the rownames
rownames(orderprop) <- order.t[, 1] # Set the rownames
rownames(familyprop) <- family.t[, 1] # Set the rownames
rownames(genusprop) <- genus.t[, 1] # Set the rownames

# proportions with variables
otuprop.v <- read.table(file = "results/tables/phylotype/otuprop.v.tsv", header = TRUE, sep = "\t")
phylumprop.v<- read.table(file = "results/tables/phylotype/phylumprop.v.tsv", header = TRUE, sep = "\t")
classprop.v <- read.table(file = "results/tables/phylotype/classprop.v.tsv", header = TRUE, sep = "\t")
orderprop.v <- read.table(file = "results/tables/phylotype/orderprop.v.tsv", header = TRUE, sep = "\t")
familyprop.v <- read.table(file = "results/tables/phylotype/familyprop.v.tsv", header = TRUE, sep = "\t")
genusprop.v <- read.table(file = "results/tables/phylotype/genusprop.v.tsv", header = TRUE, sep = "\t")


otu.scaled.t <- scale(otu.t[,-1]) # Scale the data
p.scaled.t <- scale(phylum.t[,-1], scale = TRUE) # Scale the data
c.scaled.t <- scale(class.t[,-1], scale = TRUE) # Scale the data
o.scaled.t <- scale(order.t[,-1], scale = TRUE) # Scale the data
f.scaled.t <- scale(family.t[,-1], scale = TRUE) # Scale the data
g.scaled.t <- scale(genus.t[,-1], scale = TRUE) # Scale the data

otu.scaled.prop <- scale(otuprop[,-1]) # Scale the data
p.scaled.prop <- scale(phylumprop, scale = TRUE) # Scale the data
c.scaled.prop <- scale(classprop, scale = TRUE) # Scale the data
o.scaled.prop <- scale(orderprop, scale = TRUE) # Scale the data
f.scaled.prop <- scale(familyprop, scale = TRUE) # Scale the data
g.scaled.prop <- scale(genusprop, scale = TRUE) # Scale the data


save(class, class.a, class.t, class.t.v, classprop, classprop.v, family, family.a, family.t, family.t.v, familyprop, familyprop.v, genus, genus.a, genus.t, genus.t.v, genusprop, genusprop.v, order, order.a, order.t, order.t.v, orderprop, orderprop.v, otu, otu.t, otu.t.v, otuprop, otuprop.v, phylum, phylum.a, phylum.t, phylum.t.v,	phylumprop,	phylumprop.v, otu.scaled.t, p.scaled.t, c.scaled.t, o.scaled.t, f.scaled.t, g.scaled.t, otu.scaled.prop, p.scaled.prop, c.scaled.prop, o.scaled.prop, f.scaled.prop, g.scaled.prop, file = "./results/taxonomydata.RData")
phylum
```

```{r}
load(file = "./results/taxonomydata.RData")
```

<br>

```{r message=FALSE, warning=FALSE}
library(vegan)

spaexact <- specaccum(otu.t[,-1], method="exact")
p_spa <- specaccum(phylum.t[, -1], method="exact")
c_spa <- specaccum(class.t[, -1], method="exact")
o_spa <- specaccum(order.t[, -1], method="exact")
f_spa <- specaccum(family.t[, -1], method="exact")
g_spa <- specaccum(genus.t[, -1], method="exact")

sparandom <- specaccum(otu.t[,-1], method="random")
p_spr <- specaccum(phylum.t[, -1], method="random")
c_spr <- specaccum(class.t[, -1], method="random")
o_spr <- specaccum(order.t[, -1], method="random")
f_spr <- specaccum(family.t[, -1], method="random")
g_spr <- specaccum(genus.t[, -1], method="random")

spacollector<- specaccum(otu.t[,-1], method="collector")
p_spc <- specaccum(phylum.t[, -1], method="collector")
c_spc <- specaccum(class.t[, -1], method="collector")
o_spc <- specaccum(order.t[, -1], method="collector")
f_spc <- specaccum(family.t[, -1], method="collector")
g_spc <- specaccum(genus.t[, -1], method="collector")

spararefy <- specaccum(otu.t[,-1], method="rarefaction")
p_sprf <- specaccum(phylum.t[, -1], method="rarefaction")
c_sprf <- specaccum(class.t[, -1], method="rarefaction")
o_sprf <- specaccum(order.t[, -1], method="rarefaction")
f_sprf <- specaccum(family.t[, -1], method="rarefaction")
g_sprf <- specaccum(genus.t[, -1], method="rarefaction")

```

<br>

### Individual species accumulation curves using different methods

```{r sppaccum_curve, fig.height=12, fig.width=12, message=FALSE, warning=FALSE}

par(cex.axis=1, cex.lab=1.4, cex.main=1.5, cex.sub=.8)
par(mfrow = c(6,4))

plot(spaexact, xlab="", ylab="OTU richness", col=6, lwd=2)
mtext("Exact", cex=1.2, adj = "las", line = .5)
plot(sparandom, xlab="", ylab="", col=3, lwd=2)
mtext("Random", cex=1.2, adj = "las", line = .5)
plot(spacollector, xlab="", ylab="", col=4, lwd=2)
mtext("Collector", cex=1.2, adj = "las", line = .5)
plot(spararefy, xlab="", ylab="", col="#E7B800", lwd=2)
mtext("Rarefaction", cex=1.2, adj = "las", line = .5)


plot(p_spa, xlab="", ylab="Phyla richness", col=6, lwd=2)
mtext("", cex=.8, adj = 0)
plot(p_spr, xlab="", ylab="", col=3, lwd=2)
mtext("", cex=.8, adj = 0)
plot(p_spc, xlab="", ylab="", col=4, lwd=2)
mtext("", cex=.8, adj = 0)
plot(p_sprf, xlab="", ylab="", col="#E7B800", lwd=2)
mtext("", cex=.8, adj = 0)


plot(c_spa, xlab="", ylab="Class richness", col=6, lwd=2)
mtext("", cex=.8, adj = 0)
plot(c_spr, xlab="", ylab="", col=3, lwd=2)
mtext("", cex=.8, adj = 0)
plot(c_spc, xlab="", ylab="", col=4, lwd=2)
mtext("", cex=.8, adj = 0)
plot(c_sprf, xlab="", ylab="", col="#E7B800", lwd=2)
mtext("", cex=.8, adj = 0)

plot(o_spa, xlab="", ylab="Order richness", col=6, lwd=2)
mtext("", cex=.8, adj = 0)
plot(o_spr, xlab="", ylab="", col=3, lwd=2)
mtext("", cex=.8, adj = 0)
plot(o_spc, xlab="", ylab="", col=4, lwd=2)
mtext("", cex=.8, adj = 0)
plot(o_sprf, xlab="", ylab="", col="#E7B800", lwd=2)
mtext("", cex=.8, adj = 0)


plot(f_spa, xlab="", ylab="Family richness", col=6, lwd=2)
mtext("", cex=.8, adj = 0)
plot(f_spr, xlab="", ylab="", col=3, lwd=2)
mtext("", cex=.8, adj = 0)
plot(f_spc, xlab="", ylab="", col=4, lwd=2)
mtext("", cex=.8, adj = 0)
plot(f_sprf, xlab="", ylab="", col="#E7B800", lwd=2)
mtext("", cex=.8, adj = 0)


plot(g_spa, xlab="", ylab="Genera richness", col=6, lwd=2)
mtext("", cex=.8, adj = 0)
plot(g_spr, xlab="", ylab="", col=3, lwd=2)
mtext("", cex=.8, adj = 0)
plot(g_spc, xlab="", ylab="", col=4, lwd=2)
mtext("", cex=.8, adj = 0)
plot(g_sprf, xlab="", ylab="", col="#E7B800", lwd=2)
mtext("", cex=.8, adj = 0)

mtext("Number sampled", cex = 1.2, side = 1, line = -2, outer = TRUE)
```

<br>

### Overplot graph of species accumulation curves

```{r specaccumsummary, fig.height=6, fig.width=10, message=FALSE, warning=FALSE}
cex <- par(cex.axis=1.2, cex.lab=1.4, cex.main=1.5, cex.sub=.8)
par(mfrow = c(2,3))

speciesacum <- plot(spaexact, xlab="", ylab="Number of OTUs", col=6, lwd=2)
plot(sparandom, add=TRUE, col=3, lwd=2)
plot(spararefy, add=TRUE, col="#E7B800", lwd=2)
plot(spacollector, add=TRUE, col=4, lwd=2)
mtext("A", font = 2, adj  = 0, line = 1, cex = 1)

# par(mfrow=c(2,2))
plot(p_spa, xlab="", ylab="Species richness", col=6, lwd=2)
plot(p_spr, add=TRUE, col=3, lwd=2)
plot(p_sprf, add=TRUE, col="#E7B800", lwd=2) 
plot(p_spc, add=TRUE, col=4, lwd=2) 
mtext("B", font = 2, adj  = 0, line = 1, cex = 1)

# par(mfrow=c(2,2))
plot(c_spa, xlab="", ylab="Species richness", col=6, lwd=2)
plot(c_spr, add=TRUE, col=3, lwd=2)
plot(c_sprf, add=TRUE, col="#E7B800", lwd=2)
plot(c_spc, add=TRUE, col=4, lwd=2)
mtext("C", font = 2, adj  = 0, line = 1, cex = 1)

# par(mfrow=c(2,2))
plot(o_spa, xlab="", ylab="Species richness", col=6, lwd=2)
plot(o_spr, add=TRUE, col=3, lwd=2)
plot(o_sprf, add=TRUE, col="#E7B800", lwd=2)
plot(o_spc, add=TRUE, col=4, lwd=2)
mtext("D", font = 2, adj  = 0, line = 1, cex = 1)

# par(mfrow=c(2,2))
plot(f_spa, xlab="", ylab="Species richness", col=6, lwd=2)
plot(f_spr, add=TRUE, col=3, lwd=2)
plot(f_sprf, add=TRUE, col="#E7B800", lwd=2)
plot(f_spc, add=TRUE, col=4, lwd=2)
mtext("E", font = 2, adj  = 0, line = 1, cex = 1)

# par(mfrow=c(2,2))
plot(g_spa, xlab="", ylab="Species richness", col=6, lwd=2)
plot(g_spr, add=TRUE, col=3, lwd=2)
plot(g_sprf, add=TRUE, col="#E7B800", lwd=2)
plot(g_spc, add=TRUE, col=4, lwd=2)
mtext("F", font = 2, adj  = 0, line = 1, cex = 1)

mtext("Number sampled", cex = 1.2, side = 1, line = -2, outer = TRUE)

# dim(otu.t)
# dim(phylum.t)
# dim(class.t)
# dim(order.t)
# dim(family.t)
# dim(genus.t)

```

<br>

## Rarefaction and Extrapolation (R/E)

<hr>

```{r inext_qo, eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
library(iNEXT)

otu_inextrichness <- iNEXT::iNEXT(otu[,c(7, 22, 181, 196)],  q=0, datatype="abundance", endpoint = 15000)
p_inextrichness <- iNEXT::iNEXT(phylum[,c(7, 22, 181, 196)],  q=0, datatype="abundance", endpoint = 15000)
c_inextrichness <- iNEXT::iNEXT(class[,c(7, 22, 181, 196)],  q=0, datatype="abundance", endpoint = 15000)
o_inextrichness <- iNEXT::iNEXT(order[,c(7, 22, 181, 196)],  q=0, datatype="abundance", endpoint = 15000)
f_inextrichness <- iNEXT::iNEXT(family[,c(7, 22, 181, 196)],  q=0, datatype="abundance", endpoint = 15000)
g_inextrichness <- iNEXT::iNEXT(genus[,c(7, 22, 181, 196)],  q=0, datatype="abundance", endpoint = 15000)

otu_inextshannon <- iNEXT::iNEXT(otu[,c(7, 22, 181, 196)],  q=1, datatype="abundance", endpoint = 15000)
p_inextshannon <- iNEXT::iNEXT(phylum[,c(7, 22, 181, 196)],  q=1, datatype="abundance", endpoint = 15000)
c_inextshannon <- iNEXT::iNEXT(class[,c(7, 22, 181, 196)],  q=1, datatype="abundance", endpoint = 15000)
o_inextshannon <- iNEXT::iNEXT(order[,c(7, 22, 181, 196)],  q=1, datatype="abundance", endpoint = 15000)
f_inextshannon <- iNEXT::iNEXT(family[,c(7, 22, 181, 196)],  q=1, datatype="abundance", endpoint = 15000)
g_inextshannon <- iNEXT::iNEXT(genus[,c(7, 22, 181, 196)],  q=1, datatype="abundance", endpoint = 15000)

otu_inextsimpson <- iNEXT::iNEXT(otu[,c(7, 22, 181, 196)],  q=2, datatype="abundance", endpoint = 15000)
p_inextsimpson <- iNEXT::iNEXT(phylum[,c(7, 22, 181, 196)],  q=2, datatype="abundance", endpoint = 15000)
c_inextsimpson <- iNEXT::iNEXT(class[,c(7, 22, 181, 196)],  q=2, datatype="abundance", endpoint = 15000)
o_inextsimpson <- iNEXT::iNEXT(order[,c(7, 22, 181, 196)],  q=2, datatype="abundance", endpoint = 15000)
f_inextsimpson <- iNEXT::iNEXT(family[,c(7, 22, 181, 196)],  q=2, datatype="abundance", endpoint = 15000)
g_inextsimpson <- iNEXT::iNEXT(genus[,c(7, 22, 181, 196)],  q=2, datatype="abundance", endpoint = 15000)

save(otu_inextrichness,	p_inextrichness,	c_inextrichness,	o_inextrichness,	f_inextrichness,	g_inextrichness,	otu_inextshannon,	p_inextshannon,	c_inextshannon,	o_inextshannon,	f_inextshannon,	g_inextshannon,	otu_inextsimpson,	p_inextsimpson,	c_inextsimpson,	o_inextsimpson,	f_inextsimpson,	g_inextsimpson, file = "./results/iNEXT.RData")

```


### Type 1: Sample-size-based R/E curve

```{r inexttyp1, fig.height=6, fig.width=10, message=FALSE, warning=FALSE}
library(iNEXT)
library(ggplot2)
library(scales)

otu_type1 <- ggiNEXT(otu_inextrichness,  type=1, se=FALSE, color="site") + 
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("A") + labs(x="Number sampled", y="OTU richness") + theme_grey() + normalaxislayout3
# otu_type1

p_type1 <- ggiNEXT(p_inextrichness,  type=1,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Phyla richness") + theme_grey() + normalaxislayout3 
# otu_type1

c_type1 <- ggiNEXT(c_inextrichness,  type=1,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Class richness") + theme_grey() + normalaxislayout3 
# c_type1

o_type1 <- ggiNEXT(o_inextrichness,  type=1,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Order richness") + theme_grey() + normalaxislayout3 
# o_type1

f_type1 <- ggiNEXT(f_inextrichness,  type=1,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Family richness") + theme_grey() + normalaxislayout3 
# f_type1

g_type1 <- ggiNEXT(g_inextrichness,  type=1,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Genera richness") + theme_grey() + normalaxislayout3 
# g_type1

ggarrange(otu_type1, p_type1, c_type1,o_type1,f_type1,g_type1, ncol =3, nrow = 2, common.legend = TRUE, legend = "none", align = "hv", labels = "AUTO")

```


<br>

### Type 2: Sample completeness curve

```{r inexttyp2, fig.height=6, fig.width=10, message=FALSE, warning=FALSE}
library(iNEXT)
library(ggplot2)

otu_type2 <- ggiNEXT(otu_inextrichness,  type=2, se=FALSE, color="site") + 
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("B") + labs(x="Number sampled", y="Sample coverage") + theme_grey() + normalaxislayout3
# otu_type2

p_type2 <- ggiNEXT(p_inextrichness,  type=2, se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Sample coverage") + theme_grey() + normalaxislayout3 
# otu_type2

c_type2 <- ggiNEXT(c_inextrichness,  type=2, se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Sample coverage") + theme_grey() + normalaxislayout3 
# c_type2

o_type2 <- ggiNEXT(o_inextrichness,  type=2, se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Sample coverage") + theme_grey() + normalaxislayout3 
# o_type2

f_type2 <- ggiNEXT(f_inextrichness,  type=2, se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Sample coverage") + theme_grey() + normalaxislayout3 
# f_type2

g_type2 <- ggiNEXT(g_inextrichness,  type=2, se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Number sampled", y="Sample coverage") + theme_grey() + normalaxislayout3 
# g_type2

ggarrange(otu_type2, p_type2, c_type2,o_type2,f_type2,g_type2, ncol =3, nrow = 2, common.legend = TRUE, legend = "right", align = "hv", labels = "AUTO")

```

Figure x: Species diversity estimates as a function of sample size. Only species with abundance greater or equal to 1 are detected in the sample.

<br>

### Type 3: Coverage‐based R/E curves

<hr>

```{r inexttyp3, fig.height=6, fig.width=10, message=FALSE, warning=FALSE}
library(iNEXT)
library(ggplot2)

otu_type3 <- ggiNEXT(otu_inextrichness,  type=3,  se=FALSE, color="site") + 
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("C") + labs(x="Sample coverage", y="OTU richness") + theme_grey() + normalaxislayout3
# otu_type3

p_type3 <- ggiNEXT(p_inextrichness,  type=3,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Sample coverage", y="Phyla richness") + theme_grey() + normalaxislayout3 
# otu_type3

c_type3 <- ggiNEXT(c_inextrichness,  type=3,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Sample coverage", y="Class richness") + theme_grey() + normalaxislayout3 
# c_type3

o_type3 <- ggiNEXT(o_inextrichness,  type=3,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Sample coverage", y="Order richness") + theme_grey() + normalaxislayout3 
# o_type3

f_type3 <- ggiNEXT(f_inextrichness,  type=3,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Sample coverage", y="Family richness") + theme_grey() + normalaxislayout3 
# f_type3

g_type3 <- ggiNEXT(g_inextrichness,  type=3,  se = FALSE, color="site") +
  scale_shape_manual(values=c(15,17,18,19)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend + ggtitle("") + labs(x="Sample coverage", y="Genera richness") + theme_grey() + normalaxislayout3 
# g_type3

ggarrange(otu_type3, p_type3, c_type3,o_type3,f_type3,g_type3, ncol =3, nrow = 2, common.legend = TRUE, legend = "right", align = "hv", labels = "AUTO")

```

<br>

```{r inextspeciesrich, fig.height=14, fig.width=10}
ggpubr::ggarrange(otu_type1, otu_type2, otu_type3, p_type1, p_type2, p_type3, c_type1, c_type2, c_type3, o_type1, o_type2, o_type3, f_type1, f_type2, f_type3, g_type1, g_type2, g_type3, ncol =3, nrow = 6, common.legend = TRUE, legend = "none", align = "hv", labels = "")
```

Figure x. Rarefaction and extrapolation curves. Sample-size-based curve (**A**), sample completeness curve (**B**), Coverage-based curves (**C**).

<br>

### Shannon diversity index

```{r shannon, fig.height=6, fig.width=10, warning=FALSE}

otu_shannon <- ggiNEXT(otu_inextshannon, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + ggtitle("") + bottomlegend +
  labs(x="Number sampled", y="Shannon index") + cleanup
# otu_shannon 

p_shannon <- ggiNEXT(p_inextshannon, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="Shannon index") + cleanup
# p_shannon

c_shannon <- ggiNEXT(c_inextshannon, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="Shannon index") + cleanup
# otu_shannon

o_shannon <- ggiNEXT(o_inextshannon, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="Shannon index") + cleanup

f_shannon <- ggiNEXT(f_inextshannon, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="Shannon index") + cleanup
# f_shannon


g_shannon <- ggiNEXT(g_inextshannon, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="Shannon index") + cleanup
# g_shannon

ggarrange(otu_shannon, p_shannon, c_shannon,o_shannon,f_shannon,g_shannon, ncol =3, nrow = 2, common.legend = TRUE, legend = "right", align = "hv", labels = "AUTO")

```

<br>

### Inverse Simpson diversity index

```{r invsimpson, fig.height=6, fig.width=10, warning=FALSE}
otu_invsimpson <- ggiNEXT(otu_inextsimpson, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="InvSimpson index") + ggtitle("") + cleanup
# otu_simpson

p_invsimpson <- ggiNEXT(p_inextsimpson, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="InvSimpson index") + cleanup
# p_simpson

c_invsimpson <- ggiNEXT(c_inextsimpson, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="InvSimpson index") + cleanup
# c_simpson

o_invsimpson <- ggiNEXT(o_inextsimpson, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="InvSimpson index") + cleanup
# o_simpson

f_invsimpson <- ggiNEXT(f_inextsimpson, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="InvSimpson index") + cleanup
# f_simpson

g_invsimpson <- ggiNEXT(g_inextsimpson, type=1,  se = FALSE, color="site")+
  scale_shape_manual(values=c(15,15,15,15)) + theme_bw(base_size=12)+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  scale_y_continuous(labels = comma, breaks=pretty_breaks(n=5)) + bottomlegend+
  labs(x="Number sampled", y="InvSimpson index") + cleanup
# g_simpson


ggarrange(otu_invsimpson, p_invsimpson, c_invsimpson,o_invsimpson,f_invsimpson,g_invsimpson, ncol =3, nrow = 2, common.legend = TRUE, legend = "right", align = "hv", labels = "AUTO")


```

```{r inextdiverity, eval=FALSE, fig.height=16, fig.width=10, include=FALSE}
ggarrange(otu_shannon,  otu_invsimpson, p_shannon, p_invsimpson, c_shannon, c_invsimpson, o_shannon, o_invsimpson, f_shannon, f_invsimpson, g_shannon, g_invsimpson, ncol =2, nrow = 6, common.legend = TRUE, legend = "right", align = "hv", labels = c("OTUs","", "Phylum","",  "Class","",  "Order","",  "Family","",  "Genus",""))
```


<br>

<br>

## Beta diversity

<hr>

```{r child='clustering.Rmd'}

```

<br>

## Phylogenetic clustering and tree annotation {-}

<hr>


![](img/phyloclusters.png)

Figure x: Sample Phylip or Newick-formatted tree clustered using the UPGMA (Unweighted Pair Group Method with Arithmetic Mean) algorithm. Similar data was used to construct different types of tree including rectangular (**A**), circular (**B**) and unrooted (**C**) to view how samples were clustered. 





<br>

<!-- # Data-driven Hypotheses {-} -->

<!-- Microbiome analysis can yield results in bulky that need to be highly curated to answer some specific research questions. In this example we wanted to annotate most microbes found in gut using mouse as a model (this case study). Literature have reported Firmicutes, Bacteroidetes, Actinobacteria, and Proteobacteria as the most dominant bacterial phyla found in the human gut [@Schloss2012; @Lagkouvardos2016]. The corresponding genera includes Bacteroides, Clostridium, Faecalibacterium, Eubacterium, Ruminococcus, Peptococcus, Peptostreptococcus, Bifidobacterium, Escherichia and Lactobacillus. We used phylogenetic annotation to show distibution of 15 colon bacterial genera as related to sex and time of weaning variables. -->

<!-- <br> -->

<!-- 1 *Bacteroides* -->
<!-- 2 *Clostridium* -->
<!-- 3 *Faecalibacterium* -->
<!-- 4 *Eubacterium* -->
<!-- 5 *Ruminococcus* -->
<!-- 6 *Peptococcus* -->
<!-- 7 *Peptostreptococcus* -->
<!-- 8 *Bifidobacterium* -->
<!-- 9 *Escherichia* -->
<!-- 10 *Lactobacillus* -->
<!-- 11 *Enterobacter* -->
<!-- 12 *Enterococcus* -->
<!-- 13 *Klebsiella* -->
<!-- 14 *Proteus* -->
<!-- 15 *Pseudomonas* -->
<!-- 16 *Salmonella* -->
<!-- 17 *Staphylococcus* -->


<!-- <hr> -->

<!-- ## Common bacterial genera in murine gut {-} -->

<!-- ```{r colonbacteria, include=FALSE} -->
<!-- data <- genus -->
<!-- colongeneranames <- c("Bacteroides",	"Clostridium",	"Faecalibacterium",	"Eubacterium",	"Ruminococcus",	"Peptococcus",	"Peptostreptococcus",	"Bifidobacterium",	"Escherichia",	"Lactobacillus",	"Enterobacter",	"Enterococcus",	"Klebsiella",	"Proteus",	"Pseudomonas",	"Salmonella",	"Staphylococcus") -->

<!-- colongenera <- filter(genus, colongeneranames %in% genus$taxon) -->
<!-- dim(colongenera) -->

<!-- write.table(colongenera,"results/tables/colongenera.tsv", append = FALSE, quote = FALSE, sep="\t", row.names = FALSE, col.names = TRUE, qmethod = c("escape", "double"), fileEncoding = "") -->

<!-- ``` -->

<!-- ### Colon genera identified by Phylotype approach -->
<!-- ```{r, message=FALSE, warning=FALSE, comment=NA, include=FALSE} -->
<!-- # colongenera_rel -->
<!-- library(dplyr) -->
<!-- library(kableExtra) -->
<!-- knitr::kable(tail(colongenera_rel[,c(1,2,4)],11), "html", booktabs = TRUE) %>% -->
<!-- kable_styling(bootstrap_options = "striped", font_size = 11) -->

<!-- ``` -->

<!-- ### Colon genera identified by OTU approach -->
<!-- ```{r, message=FALSE, warning=FALSE, comment=NA, include=FALSE} -->
<!-- # colongenera_rel2 -->

<!-- library(kableExtra) -->
<!-- knitr::kable(tail(colongenera_rel2[,c(1,2,4)],11), "html", booktabs = TRUE) %>% -->
<!--   kable_styling(bootstrap_options = "striped", font_size = 11) -->

<!-- ``` -->

<br>

# Posible questions {-}

<hr>

### Alpha diversity 
* QN1: Are the values obtained too sensitive to sampling? 
* QN2: Was the sampling effort sufficient to account for most OTUs present in a sample?
* QN3: Is there a need to continue with re-sampling?
* QN4: .......?

<br>


### Beta diversity 
* QN1: .......?
* QN2: .......?
* QN3: .......?
* QN4: .......?

<br>


## --------------------------------------------------------
# More intervention by investigators
* Testing hypotheses statistically ......
* Conducting biological-based analysis and intepretation ......
* Conducting focused annotation ......
* ..........
* ..........


