---
title: "Progress report 3"
bibliography: ../../library/imap.bib
date: 'Updated: `r Sys.time()`'
output:
  html_document:
    css: ../../code/css/report.css
    # keep_md: yes
link-citations: yes
subtitle: 'Sequence Processing and classification'
biblio-style: apalike
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.path='./images/', fig.show='asis',  dev = 'png', fig.align='center', out.width = "80%",  out.height = "80%", echo=TRUE, comment = NA, warning=FALSE, message=FALSE)
# knitr::opts_chunk$set(fig.path='./images/pdf/', fig.show='asis',  dev = 'pdf', fig.align='center', out.width = "80%",  out.height = "80%", echo=TRUE, comment = NA, warning=FALSE, message=FALSE)
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(comment = NA)
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning = FALSE)
load("./Rpackages.RData")
load("./functions.RData")
```

```{r message=FALSE, warning=FALSE}

samplemetadata <- readr::read_table2(file="../../resources/metadata/samplemetadata.tsv", col_names = T)
colnames(samplemetadata)[1] <- "SampleID"

# Subsetting metadata for mapping to output from dowmstream analysis
library(dplyr)
variables <- samplemetadata %>% dplyr::select("SampleID", "origID", "Sex", "Time", "DayID", "DPW")
names(variables)[2] <- "origID"
```


<br>


```{r mergereportdata, message=FALSE, warning=FALSE, include=FALSE}
library(readr)
contigrpt <- readr::read_table2(file = "../../data/mothur/qced.contigs.report")
alignrpt <- readr::read_table2(file = "../../data/mothur/qced.trim.contigs.good.unique.align.report")
save(contigrpt, alignrpt, file = "../../results/mergealignreport.RData")
```


```{r alignreport, include=FALSE}
load("../../results//mergealignreport.RData")
contigrpt_2 <- contigrpt[c(2,3,6)]
names(contigrpt_2) <- c("SeqLength","Overlap_Length","MisMatches")

write.table(contigrpt_2,"../../results/tables/contigrpt_2.tsv", append = FALSE, quote = FALSE, sep="\t", row.names = FALSE, col.names = TRUE, qmethod = c("escape", "double"), fileEncoding = "")


alignrpt_2 <- alignrpt[c(2,12,16)]
names(alignrpt_2) <- c("QueryLength","PairwiseAlignmentLength","SimBtwnQuery.Template")

write.table(alignrpt_2,"../../results/tables/alignrpt_2.tsv", append = FALSE, quote = FALSE, sep="\t", row.names = FALSE, col.names = TRUE, qmethod = c("escape", "double"), fileEncoding = "")

```

```{r, fig.height=6, fig.width=10, include=FALSE}
p1 <- funModeling::plot_num(contigrpt_2)
p2 <- funModeling::plot_num(alignrpt_2)
```

<br>

## Features of assembled sequences 
```{r fig.height=3, fig.width=8}
p1
```


<br>

## Descriptive statistics of assembled sequence
```{r message=FALSE, warning=FALSE}

knitr::kable(summary(contigrpt_2), "html", position = "center", booktabs = TRUE)

```

<br>

## Features of aligned sequences 

```{r fig.height=3, fig.width=8}
p2
```

<br>

## Descriptive statistics of aligned sequence

```{r message=FALSE, warning=FALSE}
knitr::kable(summary(alignrpt_2))

```

## Number of processed sequences 
```{bash}
paste -d'\t' \
../../data/mothur/phylotype/00_processed_seqs/qced.contigs.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.good.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count.summary \
../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count.summary \
>../../data/mothur/phylotype/00_processed_seqs/seqdepth.tsv


# paste -d'\t' ../../data/mothur/phylotype/00_processed_seqs/qced.contigs.count.summary ../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.count.summary ../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.good.count.summary  ../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.count.summary ../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count.summary ../../data/mothur/phylotype/00_processed_seqs/qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count.summary >../../data/mothur/phylotype/00_processed_seqs/seqdepth.tsv

echo -e "SampleID\tOriginal\tScreened\tAligned\tDenoised\tNonChimeric\tBacteriaOnly\tNoMock" >../../results/tables/processedSeqs.tsv
cut -f1,2,4,6,8,10,12,14 ../../data/mothur/phylotype/00_processed_seqs/seqdepth.tsv >>../../results/tables/processedSeqs.tsv

head ../../results/tables/processedSeqs.tsv

```

<br>

## Summary of sequence length in eaach sample

```{bash}

paste -d'\t' \
../../data/mothur/qced.trim.contigs.summary \
../../data/mothur/qced.trim.contigs.good.unique.summary \
../../data/mothur/qced.trim.contigs.good.unique.good.filter.unique.summary \
../../data/mothur/qced.trim.contigs.good.unique.good.filter.unique.precluster.summary \
../../data/mothur/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.summary \
../../data/mothur/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.summary \
../../data/mothur/phylotype/qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.summary \
>../../data/mothur/phylotype/00_processed_seqs/seqsummary.tsv

echo -e "Original\tScreened\tAligned\tDenoised\tNonChimeric\tBacteriaOnly\tNoMock" >../../results/tables/seqsummary.tsv
cut -f4,11,18,25,32,39,46 ../../data/mothur/phylotype/00_processed_seqs/seqsummary.tsv >>../../results/tables/seqsummary.tsv

head ../../results/tables/seqsummary.tsv

```

## Inspect sequence length
```{r}
library(ggplot2)
seqlen <- readr::read_table2("../../results/tables/seqsummary.tsv")
seqlen <- seqlen[-1,]

seqlen
```




```{r processedseqdata, message=FALSE, warning=FALSE}
library(dplyr)
library(readr)
processedseqs <- read_table2(file="../../results/tables/processedSeqs.tsv", col_names = TRUE)
# names(processedseqs)
# dim(processedseqs)

processedseqs <- processedseqs[-c(9:10),] # Remove Mock rows. Be sure to remove only the rows that contains the Mock data

seqcount.v <- cbind(variables, processedseqs[,-2])
names(seqcount.v)[7] <- "Group"
# funModeling::df_status(seqcount.v)

```

<br>

## Summary statistics of processed sequences
```{r}
# names(seqcount.v[,8:13])
summary(seqcount.v[,8:13])

```

<br>

## Distribution of processed sequences {-}

<hr>


```{r}
seqcount.v.m <- reshape2::melt(seqcount.v, id = c("SampleID", "origID", "Sex", "Time", "DayID", "DPW", "Group"))

# seqcount.v.m <- seqcount.v.m[-c(41:48),]
seqcount.v.m$value <- as.numeric(seqcount.v.m$value)
```

<br>

### Stacked barplot {-}

```{r seqbar1, fig.height=6, fig.width=10, message=FALSE, warning=FALSE}

library(ggplot2)
library(ggpubr)

seqcount.v.m$SampleID <- as.factor(seqcount.v.m$SampleID)
seqcount.v.m$DPW <- as.factor(seqcount.v.m$DPW)
seqcount.v.m$value <- as.numeric(seqcount.v.m$value)
seqcount.v.m$variable <- as.factor(seqcount.v.m$variable)

seqbar1 <- ggplot(seqcount.v.m, aes(x = reorder(SampleID, -value), y = value, fill = variable)) +
geom_bar(stat = "identity") +
labs(x = "Data Point", y = "Sequence depth", title = "") +
theme(axis.text=element_text(size=10))

ggpubr::ggarrange(seqbar1, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "right", align = "hv", labels = "")

```

Figure x: Stacked barplots of processed sequences grouped by processing variable


<br>

### Boxplots grouped by process
```{r, fig.height=6, fig.width=10,}
library(scales)

boxforintext2 <- ggboxplot(seqcount.v.m, x = "variable", y = "value",
                color = "variable", palette =c("rainbow"),
                add = "jitter", shape = 19) +  xlab("") + ylab("Sequence depth") + ggtitle("") + nolegend +
  theme(axis.text.x=element_text(size=10, angle = 0, hjust = 1)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(axis.title=element_text(size=10,face="bold"))+ cleanup+ theme(legend.position="none")+
  scale_y_continuous(breaks=pretty_breaks(n=8))
boxforintext2
```

<br>

### Boxplots grouped by sex and time
```{r boxrpt3_2, , fig.height=6, fig.width=10}

boxplot <- ggboxplot(seqcount.v.m, x = "variable", y = "value",
                color = "Sex", palette =c("rainbow"),
                add = "jitter") +  xlab("") + ylab("Sequence depth") + ggtitle("")  
ggpubr::ggarrange(boxplot, ncol = 1, nrow = 1,  common.legend = FALSE, legend = "top", align = "hv", labels = "")
```

<br>

### Density plots
```{r, fig.height=4, fig.width=8,}

sexintext_densplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "stack", alpha = 0.05,
   color = "variable", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Density") +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust = 1)) + facet_grid(~Sex) +
  theme(axis.text.y=element_text(size=10)) +
  theme(axis.title=element_text(size=10,face="bold"))+ cleanup+ theme(legend.position="right")+
  scale_x_continuous(breaks=pretty_breaks(n=8))
sexintext_densplot
```

<br>

```{r fig.height=4, fig.width=8, message=FALSE, warning=FALSE}
timeintext_densplot <- ggdensity(seqcount.v.m, x = "value",
   add = "mean", rug = TRUE, position = "dodge", alpha = 0.05,
   color = "variable", fill = "variable",
   palette = c("rainbow")) + xlab("Sequence depth") + ylab("Density") +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust = 1)) + facet_grid(~Time) +
  theme(axis.text.x=element_text(size=10, angle = 90, hjust = 1)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(axis.title=element_text(size=10,face="bold"))+ cleanup+ theme(legend.position="right")+
  scale_x_continuous(breaks=pretty_breaks(n=8))
timeintext_densplot
```


<br>

# Posible questions {-}

<hr>

### At sequence processing 
* QN1: Do the paired sequences overlap as expected?
* QN2: Are there sequences to be removed from the analysis based on sequence length?
* QN3: Does the dorminant length match the targeted region of 16S rRNA gene?
* QN4: Are the representative sequences too few or too many? <mark>Note that too many non-redundant sequences may results from poor overlapping between forward and reverse reads. Review before proceeding to avoid misleading conclusions<mark>.


<br>

### At sequence classification {-}
* QN1: What taxonomic classifier should be used? Think of need to develop and use custom classifier that suits the study objectives.
* QN2: Are there sequences assigned to non-bacterial taxonomic lineages?
* QN3: Should the matches to non-bacterial lineages be removed from further analysis?
* QN4: .......?

<br>

### At error estimation {-}
* QN1: Is the error too high to cause rejection of the results results?
* QN2: .......?
* QN3: .......?
* QN4: .......?


<br>

## Summary of packages used in the analysis

<hr>

```{r}
sessionInfo()
```

<br>

